Index: assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java
===================================================================
--- assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java	(revision 12001)
+++ assignment/assignment-impl/impl/src/java/org/sakaiproject/assignment/impl/BaseAssignmentService.java	(revision 12002)
@@ -2935,7 +2935,7 @@
 								{
 									submittersString = submittersString.concat("; ");
 								}
-								submittersString = submittersString.concat(submitters[i].getSortName());
+								submittersString = submittersString.concat(submitters[i].getLastName()+","+submitters[i].getFirstName());
 							}
 							submittersName = submittersName.concat(submittersString);
 							submittedText = s.getSubmittedText();
