Index: content/content-tool/.classpath
===================================================================
--- content/content-tool/.classpath	(revision 12246)
+++ content/content-tool/.classpath	(revision 12247)
@@ -19,5 +19,6 @@
 	<classpathentry combineaccessrules="false" kind="src" path="/metaobj"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/util-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/util-util"/>
+	<classpathentry combineaccessrules="false" kind="src" path="/triples-api"/>
 	<classpathentry kind="output" path="bin"/>
 </classpath>
Index: content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java
===================================================================
--- content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java	(revision 12246)
+++ content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java	(revision 12247)
@@ -7139,23 +7139,33 @@
 		Collection groupRefs = new TreeSet();
 		if(this.m_allowGroupResources)
 		{
-			ContentCollection container;
+			ContentEntity entity;
 			try
 			{
 				Reference ref = m_entityManager.newReference(refString);
 				Site site = m_siteService.getSite(ref.getContext());
 				
-				container = findCollection(ref.getId());
-				if(AccessMode.INHERITED == container.getAccess())
+				if(ref.getId().endsWith(Entity.SEPARATOR))
 				{
-					groups.addAll(container.getInheritedGroupObjects());
-					groupRefs.addAll(container.getInheritedGroups());
+					entity = findCollection(ref.getId());
 				}
 				else
 				{
-					groups.addAll(container.getGroupObjects());
-					groupRefs.addAll(container.getGroups());
+					entity = findResource(ref.getId());
 				}
+				if(entity != null)
+				{
+					if(AccessMode.INHERITED == entity.getAccess())
+					{
+						groups.addAll(entity.getInheritedGroupObjects());
+						groupRefs.addAll(entity.getInheritedGroups());
+					}
+					else
+					{
+						groups.addAll(entity.getGroupObjects());
+						groupRefs.addAll(entity.getGroups());
+					}
+				}
 
 				if(groups.isEmpty())
 				{
@@ -7171,7 +7181,7 @@
 				{
 					rv.addAll(groups);
 				}
-				else if(SecurityService.unlock(EVENT_RESOURCE_ALL_GROUPS, site.getReference()) && unlockCheck(function, container.getId()))
+				else if(SecurityService.unlock(EVENT_RESOURCE_ALL_GROUPS, site.getReference()) && unlockCheck(function, entity.getId()))
 				{
 					rv.addAll(groups);
 				}
