Index: web/news-tool/tool/src/webapp/vm/news/chef_news-List.vm
===================================================================
--- web/news-tool/tool/src/webapp/vm/news/chef_news-List.vm	(revision 13144)
+++ web/news-tool/tool/src/webapp/vm/news/chef_news-List.vm	(revision 13145)
@@ -30,7 +30,7 @@
 						#if($item.Link)
 							 <a href="$validator.escapeHtml($item.Link)" target="_blank" title="$tlang.getString("goto.item")">($full_story)</a>
 						 #end
-						 #if($item.Pubdate)
+						 #if($item.Pubdate && !$item.Pubdate == "")
 							($item.Pubdate)
 						#end
 					</p>
