Index: content/content-impl/impl/src/java/org/sakaiproject/content/impl/CollectionAccessFormatter.java
===================================================================
--- content/content-impl/impl/src/java/org/sakaiproject/content/impl/CollectionAccessFormatter.java	(revision 17928)
+++ content/content-impl/impl/src/java/org/sakaiproject/content/impl/CollectionAccessFormatter.java	(working copy)
@@ -32,6 +32,7 @@
 import org.sakaiproject.component.cover.ServerConfigurationService;
 import org.sakaiproject.content.api.ContentCollection;
 import org.sakaiproject.content.api.ContentResource;
+import org.sakaiproject.content.cover.ContentHostingService;
 import org.sakaiproject.entity.api.Entity;
 import org.sakaiproject.entity.api.Reference;
 import org.sakaiproject.entity.api.ResourceProperties;
@@ -48,6 +50,7 @@
  */
 public class CollectionAccessFormatter
 {
+
 	/**
 	 * Format the collection as an HTML display.
 	 */
@@ -57,6 +60,18 @@
 		PrintWriter out = null;
 		// don't set the writer until we verify that
 		// getallresources is going to work.
+	
+		if(ContentHostingService.isAttachmentResource(x.getId())) 
+		{
+			try
+			{
+				res.sendError(HttpServletResponse.SC_NOT_FOUND);
+				return;
+			} catch ( java.io.IOException e ) {
+				return; 
+			}
+		}
+
 		boolean printedHeader = false;
 		boolean printedDiv = false;
 
