svn diff -r19804:19805 https://source.sakaiproject.org/svn
Index: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java
===================================================================
--- assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java	(revision 19804)
+++ assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java	(revision 19805)
@@ -532,6 +532,9 @@
 	/** For storing an instance of the TaggingManager */
 	protected TaggingManager taggingManager = (TaggingManager) ComponentManager.get("org.sakaiproject.assignment.taggable.api.TaggingManager");
 	
+	/** property for previous feedback attachments **/
+	private static final String PROP_SUBMISSION_PREVIOUS_FEEDBACK_ATTACHMENTS = "prop_submission_previous_feedback_attachments";
+	
 	/**
 	 * central place for dispatching the build routines based on the state name
 	 */
@@ -728,6 +731,11 @@
 				{
 					context.put("prevFeedbackComment", p.getProperty(ResourceProperties.PROP_SUBMISSION_PREVIOUS_FEEDBACK_COMMENT));
 				}
+				
+				if (p.getProperty(PROP_SUBMISSION_PREVIOUS_FEEDBACK_ATTACHMENTS) != null)
+				{
+					context.put("prevFeedbackAttachments", getPrevFeedbackAttachments(p));
+				}
 			}
 		}
 		catch (IdUnusedException e)
@@ -1357,6 +1365,11 @@
 				{
 					context.put("prevFeedbackComment", p.getProperty(ResourceProperties.PROP_SUBMISSION_PREVIOUS_FEEDBACK_COMMENT));
 				}
+				
+				if (p.getProperty(PROP_SUBMISSION_PREVIOUS_FEEDBACK_ATTACHMENTS) != null)
+				{
+					context.put("prevFeedbackAttachments", getPrevFeedbackAttachments(p));
+				}
 
 				if (p.getProperty(GRADE_SUBMISSION_ALLOW_RESUBMIT) != null)
 				{
@@ -1409,6 +1422,17 @@
 
 	} // build_instructor_grade_submission_context
 
+	private List getPrevFeedbackAttachments(ResourceProperties p) {
+		String attachmentsString = p.getProperty(PROP_SUBMISSION_PREVIOUS_FEEDBACK_ATTACHMENTS);
+		String[] attachmentsReferences = attachmentsString.split(",");
+		List prevFeedbackAttachments = EntityManager.newReferenceList();
+		for (int k =0; k < attachmentsReferences.length; k++)
+		{
+			prevFeedbackAttachments.add(EntityManager.newReference(attachmentsReferences[k]));
+		}
+		return prevFeedbackAttachments;
+	}
+
 	/**
 	 * build the instructor preview of grading submission
 	 */
@@ -2773,6 +2797,20 @@
 							feedbackCommentHistory = sEdit.getFeedbackComment() + "\n" + feedbackCommentHistory;
 							sPropertiesEdit.addProperty(ResourceProperties.PROP_SUBMISSION_PREVIOUS_FEEDBACK_COMMENT,
 									feedbackCommentHistory);
+							
+							// keep the history of assignment feed back comment
+							String feedbackAttachmentHistory = sPropertiesEdit
+									.getProperty(PROP_SUBMISSION_PREVIOUS_FEEDBACK_ATTACHMENTS) != null ? sPropertiesEdit
+									.getProperty(PROP_SUBMISSION_PREVIOUS_FEEDBACK_ATTACHMENTS)
+									: "";
+							List feedbackAttachments = sEdit.getFeedbackAttachments();
+							for (int k = 0; k<feedbackAttachments.size();k++)
+							{
+								feedbackAttachmentHistory = ((Reference) feedbackAttachments.get(k)).getReference() + "," + feedbackAttachmentHistory;
+							}
+							
+							sPropertiesEdit.addProperty(PROP_SUBMISSION_PREVIOUS_FEEDBACK_ATTACHMENTS,
+									feedbackAttachmentHistory);
 
 							// reset the previous grading context
 							sEdit.setFeedbackText("");
Index: assignment/assignment-tool/tool/src/bundle/assignment.properties
===================================================================
--- assignment/assignment-tool/tool/src/bundle/assignment.properties	(revision 19804)
+++ assignment/assignment-tool/tool/src/bundle/assignment.properties	(revision 19805)
@@ -263,6 +263,7 @@
 gradingsub.opecliass  = Open, click to close assignment instruction
 gradingsub.prefee     = Previous Feedback Comment
 gradingsub.prefeetex  = Previous Feedback Text
+gradingsub.prefeedbackAttachments  = Previous Returned Attachments
 gradingsub.pregra     = Previous Grade:
 gradingsub.usethebel1 = Use the box below to enter additional summary comments about this submission
 gradingsub.usethebel2 = Use the box below to enter summary comments about this submission's attachment or the grade
Index: assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_instructor_grading_submission.vm
===================================================================
--- assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_instructor_grading_submission.vm	(revision 19804)
+++ assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_instructor_grading_submission.vm	(revision 19805)
@@ -243,9 +243,51 @@
 							#end
 						</h4>
 						#set ($size = 0)
+						#set ($props = false)
+						#foreach ($attachment in $gradingAttachments) 
+							#set ($props = $attachment.Properties) 
+							#if ($!props)
+								#set ($size = $size + 1)
+							#end
+						#end
+						#if ($size == 0)
+							<p class="instruction">$tlang.getString("gen.noatt")</p>
+						#else
+							<ul class="attachList indnt1">
+								#foreach ($attachment in $gradingAttachments) 
+									#set ($props = false)
+									#set ($props = $attachment.Properties) 
+									#if ($!props)
+										<li>
+											#if ($props.getBooleanProperty($props.NamePropIsCollection))
+												<img src = "#imageLink($contentTypeImageService.getContentTypeImage("folder"))" border="0" alt="$tlang.getString("gen.folatt")" />
+											#else
+												<img src = "#imageLink($contentTypeImageService.getContentTypeImage($props.getProperty($props.NamePropContentType)))" border="0" alt="$tlang.getString("gen.filatt")" />
+											#end
+											<a href="$attachment.Url" target="_blank">$validator.escapeHtml($props.getPropertyFormatted($props.NamePropDisplayName))</a>						
+											#if (!$props.getBooleanProperty($props.NamePropIsCollection))
+												($props.getPropertyFormatted($props.NamePropContentLength))
+											#end
+										</li>
+									#end
+								#end
+							</ul>
+						#end		
+						<p class="act">
+							#if ($!props)
+								<input type="button" name="attach" value="$tlang.getString('gen.adddroatt')" onclick="javascript:document.gradeForm.onsubmit();document.getElementById('option').value='attach';document.gradeForm.submit();return false;" accesskey="a"/>
+							#else
+								<input type="button" name="attach" value="$tlang.getString('gen.addatt')" onclick="javascript:document.gradeForm.onsubmit();document.getElementById('option').value='attach';document.gradeForm.submit();return false;" accesskey="a"/>
+							#end
+						</p>
+						## show previous grading/feedback attachments if any
+						#if ($!prevFeedbackAttachments && $prevFeedbackAttachments.size()>0)
+							<h4>
+								<h4>$tlang.getString("gradingsub.prefeedbackAttachments")</h4>
+							</h4>
 							#set ($size = 0)
 							#set ($props = false)
-							#foreach ($attachment in $gradingAttachments) 
+							#foreach ($attachment in $!prevFeedbackAttachments) 
 								#set ($props = $attachment.Properties) 
 								#if ($!props)
 									#set ($size = $size + 1)
@@ -255,7 +297,7 @@
 								<p class="instruction">$tlang.getString("gen.noatt")</p>
 							#else
 								<ul class="attachList indnt1">
-									#foreach ($attachment in $gradingAttachments) 
+									#foreach ($attachment in $!prevFeedbackAttachments) 
 										#set ($props = false)
 										#set ($props = $attachment.Properties) 
 										#if ($!props)
@@ -273,14 +315,8 @@
 										#end
 									#end
 								</ul>
-							#end			
-							<p class="act">
-								#if ($!props)
-									<input type="button" name="attach" value="$tlang.getString('gen.adddroatt')" onclick="javascript:document.gradeForm.onsubmit();document.getElementById('option').value='attach';document.gradeForm.submit();return false;" accesskey="a"/>
-								#else
-									<input type="button" name="attach" value="$tlang.getString('gen.addatt')" onclick="javascript:document.gradeForm.onsubmit();document.getElementById('option').value='attach';document.gradeForm.submit();return false;" accesskey="a"/>
-								#end
-							</p>
+							#end	
+						#end	
 #*					<!-- Grade types
 						GRADE_TYPE_NOT_SET: -1; 
 						UNGRADED_GRADE_TYPE: 1; 
Index: assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_student_view_submission.vm
===================================================================
--- assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_student_view_submission.vm	(revision 19804)
+++ assignment/assignment-tool/tool/src/webapp/vm/assignment/chef_assignments_student_view_submission.vm	(revision 19805)
@@ -258,6 +258,10 @@
 					#end
 				#elseif ($submission.getReturned())
 						#if ($submission.TimeReturned.after($submission.TimeSubmitted))
+							<h4>$tlang.getString("gen.orisub2")</h4>
+							<div class="textPanel">
+								$cheffeedbackhelper.escapeAssignmentFeedback($submission.FeedbackText)
+							</div>
 							## for returned submission, show the instructor commented text as well
 							#if ($!prevFeedbackText)
 								<h4>$tlang.getString("gradingsub.prefeetex")</h4>
@@ -265,10 +269,6 @@
 									$cheffeedbackhelper.showPrevFeedback($cheffeedbackhelper.escapeAssignmentFeedback($!prevFeedbackText))
 								</div>
 							#end
-							<h4>$tlang.getString("gen.orisub2")</h4>
-							<div class="textPanel">
-								$cheffeedbackhelper.escapeAssignmentFeedback($submission.FeedbackText)
-							</div>
 							<h4>$tlang.getString("stuviewsubm.entbelo")</h4>
 							<table border="0"><tr><td>
 							<textarea cols="80" rows="15" wrap="virtual" name="$name_submission_text" id="$name_submission_text" #if(!$allowSubmit)disabled="disabled"#end >#if($!value_submission_text)$validator.escapeHtmlFormattedTextarea($!value_submission_text)#end</textarea>
@@ -387,20 +387,56 @@
 					#set ($size = 0)
 					#set ($props = false)
 					#set ($feedbackAttachments = $submission.FeedbackAttachments)	
+					#set ($size = 0)
+					#set ($props = false)
+					#foreach ($attachment in $feedbackAttachments) 
+						#set ($props = $attachment.Properties) 
+						#if ($!props)
+							#set ($size = $size + 1)
+						#end
+					#end
+					#if ($size == 0)
+						##$tlang.getString("gen.noatt")
+					#else
+						<h5>$tlang.getString("gen.addinstatts")</h5>
+						<ul class="attachList indnt1">
+							#foreach ($attachment in $feedbackAttachments) 
+								#set ($props = false)
+								#set ($props = $attachment.Properties) 
+								#if ($!props)
+									<li>
+										#if ($props.getBooleanProperty($props.NamePropIsCollection))
+											<img src = "#imageLink($contentTypeImageService.getContentTypeImage("folder"))" border="0" alt="$tlang.getString("gen.folatt")" />
+										#else
+											<img src = "#imageLink($contentTypeImageService.getContentTypeImage($props.getProperty($props.NamePropContentType)))" border="0" alt="$tlang.getString("gen.filatt")" />
+										#end
+										<a href="$attachment.Url" target="_blank">$validator.escapeHtml($props.getPropertyFormatted($props.NamePropDisplayName))</a>						
+										#if (!$props.getBooleanProperty($props.NamePropIsCollection))
+											($props.getPropertyFormatted($props.NamePropContentLength))
+										#end
+									</li>
+								#end
+							#end
+						</ul>
+					#end
+					## show previous grading/feedback attachments if any
+					#if ($!prevFeedbackAttachments && $prevFeedbackAttachments.size()>0)
+						<h4>
+							<h4>$tlang.getString("gradingsub.prefeedbackAttachments")</h4>
+						</h4>
 						#set ($size = 0)
 						#set ($props = false)
-						#foreach ($attachment in $feedbackAttachments) 
+						#foreach ($attachment in $!prevFeedbackAttachments) 
 							#set ($props = $attachment.Properties) 
 							#if ($!props)
 								#set ($size = $size + 1)
 							#end
 						#end
 						#if ($size == 0)
-							##$tlang.getString("gen.noatt")
+							<p class="instruction">$tlang.getString("gen.noatt")</p>
 						#else
-							<h5>$tlang.getString("gen.addinstatts")</h5>
 							<ul class="attachList indnt1">
-								#foreach ($attachment in $feedbackAttachments) 
+								#foreach ($attachment in $!prevFeedbackAttachments) 
 									#set ($props = false)
 									#set ($props = $attachment.Properties) 
 									#if ($!props)
@@ -418,7 +454,8 @@
 									#end
 								#end
 							</ul>
-						#end
+						#end	
+					#end	
 				#end
 				#if ($withGrade)
 					#set ($previousGrades =  $!submission.Properties.getProperty($!submission.Properties.getNamePropSubmissionScaledPreviousGrades()))
