svn diff -r13447:13448 https://source.sakaiproject.org/svn
Index: postem/postem-app/src/java/org/sakaiproject/tool/postem/PostemTool.java
===================================================================
--- postem/postem-app/src/java/org/sakaiproject/tool/postem/PostemTool.java	(revision 13447)
+++ postem/postem-app/src/java/org/sakaiproject/tool/postem/PostemTool.java	(revision 13448)
@@ -72,6 +72,8 @@
 	protected Gradebook oldGradebook;
 
 	protected String userId;
+	
+	protected String userEid;
 
 	protected String filename = null;
 
@@ -133,6 +135,14 @@
 	public ArrayList getGradebooks() {
 		if (userId == null) {
 			userId = SessionManager.getCurrentSessionUserId();
+			
+			if (userId != null) {
+				try {
+					userEid = UserDirectoryService.getUserEid(userId);
+				} catch (UserNotDefinedException e) {
+					LOG.error("UserNotDefinedException", e);
+				}
+			}
 		}
 
 		Placement placement = ToolManager.getCurrentPlacement();
@@ -222,6 +232,13 @@
 
 	public String getCurrentStudentGrades() {
 		this.userId = SessionManager.getCurrentSessionUserId();
+		if (this.userId != null) {
+			try {
+				this.userEid = UserDirectoryService.getUserEid(this.userId);
+			} catch (UserNotDefinedException e) {
+				LOG.error("UserNotDefinedException:",e);
+			}
+		}
 		/*
 		 * if (checkAccess()) { if (currentGradebook.getTemplate() != null) { return
 		 * currentGradebook.getTemplate().getTemplateCode(); } else { return "<p>No
@@ -230,10 +247,10 @@
 		if (currentGradebook == null) {
 			return "<p>" + msgs.getString("no_gradebook_selected") + "</p>";
 		}
-		if (!currentGradebook.hasStudent(this.userId)) {
+		if (!currentGradebook.hasStudent(this.userEid)) {
 			return "<p>" + msgs.getString("no_grades_for_user") + " " + currentGradebook.getTitle() + ".</p>";
 		}
-		return currentGradebook.studentGrades(this.userId).formatGrades();
+		return currentGradebook.studentGrades(this.userEid).formatGrades();
 	}
 
 	public String getFirstStudentGrades() {
