svn diff -r14190:14191 https://source.sakaiproject.org/svn
Index: rwiki/rwiki-impl/impl/src/java/uk/ac/cam/caret/sakai/rwiki/component/service/impl/RWikiObjectServiceImpl.java
===================================================================
--- rwiki/rwiki-impl/impl/src/java/uk/ac/cam/caret/sakai/rwiki/component/service/impl/RWikiObjectServiceImpl.java	(revision 14190)
+++ rwiki/rwiki-impl/impl/src/java/uk/ac/cam/caret/sakai/rwiki/component/service/impl/RWikiObjectServiceImpl.java	(revision 14191)
@@ -41,7 +41,11 @@
 import org.sakaiproject.component.cover.ServerConfigurationService;
 import org.sakaiproject.email.api.DigestService;
 import org.sakaiproject.entity.api.Entity;
+import org.sakaiproject.entity.api.EntityAccessOverloadException;
+import org.sakaiproject.entity.api.EntityCopyrightException;
 import org.sakaiproject.entity.api.EntityManager;
+import org.sakaiproject.entity.api.EntityNotDefinedException;
+import org.sakaiproject.entity.api.EntityPermissionException;
 import org.sakaiproject.entity.api.HttpAccess;
 import org.sakaiproject.entity.api.Reference;
 import org.sakaiproject.entity.api.ResourceProperties;
@@ -1261,7 +1265,11 @@
 		{
 			public void handleAccess(HttpServletRequest req,
 					HttpServletResponse res, Reference ref,
-					Collection copyrightAcceptedRefs)
+					Collection copyrightAcceptedRefs) 
+					throws EntityPermissionException, 
+					EntityNotDefinedException, 
+					EntityAccessOverloadException, 
+					EntityCopyrightException
 
 			{
 				checkReference(ref);
@@ -1307,11 +1315,15 @@
 					}
 					else
 					{
-						res.reset();
-						res.sendError(HttpServletResponse.SC_NOT_FOUND,
-								" Resource Now found " + ref.getReference());
+						throw new EntityNotDefinedException(ref.getReference());
 					}
 				}
+				catch (org.sakaiproject.exception.PermissionException p ) {
+					throw new EntityPermissionException(p.getUser(),p.getLock(),p.getResource());
+				}
+				catch ( EntityNotDefinedException e) {
+					throw e;
+				}
 				catch (Throwable t)
 				{
 					log.warn("Error getting wiki page via access :"
