svn diff -r19730:19731 https://source.sakaiproject.org/svn
Index: content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java
===================================================================
--- content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java	(revision 19730)
+++ content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java	(revision 19731)
@@ -620,7 +620,7 @@
 			this.getResourceTypeRegistry().register(new HtmlDocumentType());
 			this.getResourceTypeRegistry().register(new UrlResourceType());
 		}
-
+		
 	} // init
 
 	/**
@@ -2081,7 +2081,7 @@
 			{
 				// add a default value that sorts new items after existing items, with new folders before new resources
 				ContentCollection container = edit.getContainingCollection();
-				int count = container.getMembers().size();
+				int count = container.getMemberCount();
 				props.addProperty(ResourceProperties.PROP_CONTENT_PRIORITY, Integer.toString(count + 1));
 			}
 		}
