svn diff -r14745:14746 https://source.sakaiproject.org/svn
Index: osp/jsf/widgets/src/java/org/theospi/jsf/component/XmlDocumentComponent.java
===================================================================
--- osp/jsf/widgets/src/java/org/theospi/jsf/component/XmlDocumentComponent.java	(revision 14745)
+++ osp/jsf/widgets/src/java/org/theospi/jsf/component/XmlDocumentComponent.java	(revision 14746)
@@ -67,7 +67,12 @@
    public void setFactory(XmlTagFactory factory) {
       this.factory = factory;
    }
-
+   
+   /**
+    * Any time this is called the calling method MUST close the input stream!!
+    * This has the potential of causing memory leaks if the calling method does not close the stream
+    * @return InputStream
+    */
    public InputStream getXmlFile() {
       ValueBinding vb = getValueBinding("xmlFile");
       return (InputStream) vb.getValue(getFacesContext());
Index: osp/jsf/widgets/src/java/org/theospi/jsf/tag/XmlDocumentTag.java
===================================================================
--- osp/jsf/widgets/src/java/org/theospi/jsf/tag/XmlDocumentTag.java	(revision 14745)
+++ osp/jsf/widgets/src/java/org/theospi/jsf/tag/XmlDocumentTag.java	(revision 14746)
@@ -109,8 +109,9 @@
 
    protected UIComponent findComponent(FacesContext context) throws JspException {
        XmlDocumentComponent docComponent = (XmlDocumentComponent) super.findComponent(context);
-      if (docComponent.getXmlFile() == null){
-        return docComponent;
+       InputStream xmlFile = docComponent.getXmlFile();
+      if (xmlFile == null) {
+         return docComponent;
       }
       if (docComponent.getXmlRootComponent() != null && docComponent.getOldXmlFileId() != null) {
          String lastId = docComponent.getOldXmlFileId();
@@ -134,9 +135,7 @@
       try {
          SAXParserFactory parserFactory = SAXParserFactory.newInstance();
          parserFactory.setNamespaceAware(true);
-         InputStream xmlFile = docComponent.getXmlFile();
          parserFactory.newSAXParser().parse(xmlFile, handler);
-         xmlFile.close();
       }
       catch (SAXException e) {
          throw new JspException(e);
@@ -146,6 +145,12 @@
       }
       catch (IOException e) {
          throw new JspException(e);
+      } finally {
+         try {
+            xmlFile.close();
+         } catch (IOException e) {
+            throw new JspException(e);
+         }
       }
 
       docComponent.setXmlRootComponent(base);
Index: osp/presentation/tool/src/java/org/theospi/portfolio/presentation/tool/DecoratedPage.java
===================================================================
--- osp/presentation/tool/src/java/org/theospi/portfolio/presentation/tool/DecoratedPage.java	(revision 14745)
+++ osp/presentation/tool/src/java/org/theospi/portfolio/presentation/tool/DecoratedPage.java	(revision 14746)
@@ -20,6 +20,8 @@
 **********************************************************************************/
 package org.theospi.portfolio.presentation.tool;
 
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.Collections;
@@ -27,6 +29,8 @@
 import javax.faces.context.ExternalContext;
 import javax.faces.context.FacesContext;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 import org.sakaiproject.metaobj.shared.model.Id;
 import org.sakaiproject.tool.api.ToolSession;
 import org.sakaiproject.tool.cover.SessionManager;
@@ -46,6 +50,8 @@
  */
 public class DecoratedPage implements Comparable {
 
+   protected final transient Log logger = LogFactory.getLog(getClass());
+   
    private FreeFormTool parent;
    private PresentationPage base;
    private RegionMap regionMap;
@@ -126,17 +132,50 @@
       this.base = base;
    }
 
+   /**
+    * Any time this is called the calling method MUST close the input stream!!
+    * This has the potential of causing memory leaks if the calling method does not close the stream
+    * @return InputStream
+    */
    public InputStream getXmlFile() {
       InputStream  inputStream = null;
       if (getBase() != null && getBase().getLayout() != null){
-       Node node = getParent().getPresentationManager().getNode(
+         Node node = getParent().getPresentationManager().getNode(
             getBase().getLayout().getXhtmlFileId(), getBase().getLayout());
-      inputStream = node.getInputStream();
+         inputStream = node.getInputStream();
+         
+         // we want to read the entire file into memory so wo can close the inputStream
+         //    and thus release the database connection / file connection
+         
+         ByteArrayOutputStream bytesOS = new ByteArrayOutputStream();
+         int buffersize = 1024, s;
+         byte[] buffer = new byte[buffersize];
+         try {
+            while((s = inputStream.read(buffer)) != -1) {
+               bytesOS.write(buffer, 0, s);
+            }
+            inputStream.close();
+            inputStream = new ByteArrayInputStream(bytesOS.toByteArray());
+         } catch(IOException ioe) {
+            logger.error(ioe);
+            inputStream = null;
+         }
       }
 
       return inputStream;
    }
 
+   public boolean isXmlFileNotNull() {
+      InputStream  inputStream = getXmlFile();
+      boolean isNotNull = inputStream != null;
+      try {
+         inputStream.close();
+      } catch(IOException ioe) {
+         logger.equals(ioe);
+      }
+      return isNotNull;
+   }
+
    public String getXmlFileId() {
       return getBase().getLayout().getId().getValue() + getBase().getLayout().getModified().toString();
    }
Index: osp/presentation/tool/src/webapp/freeForm/pageProperties.jsp
===================================================================
--- osp/presentation/tool/src/webapp/freeForm/pageProperties.jsp	(revision 14745)
+++ osp/presentation/tool/src/webapp/freeForm/pageProperties.jsp	(revision 14746)
@@ -138,7 +138,7 @@
                     </h:panelGrid>
             <ospx:xmlDocument factory="#{freeForm.factory}"
                               xmlFile="#{freeForm.currentPage.xmlFile}"
-                              var="freeForm.currentPage.regionMap" rendered ="#{freeForm.currentPage.xmlFile != null}"/>
+                              var="freeForm.currentPage.regionMap" rendered ="#{freeForm.currentPage.xmlFileNotNull}"/>
         </f:subview>             
     </ospx:xheaderdrawer>
 </ospx:xheader>
