svn diff -r13334:13335 https://source.sakaiproject.org/svn
Index: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java
===================================================================
--- assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java	(revision 13334)
+++ assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java	(revision 13335)
@@ -6523,7 +6523,6 @@
 		{
 			//	get all active site users
 			String authzGroupId = SiteService.siteReference(contextString);
-			List allowAddAssignmentUsers = AssignmentService.allowAddAssignmentUsers(contextString);
 			try
 			{
 				AuthzGroup group = AuthzGroupService.getAuthzGroup(authzGroupId);
@@ -6534,11 +6533,7 @@
 					try
 					{
 						User u = UserDirectoryService.getUser(userId);
-						// only return student
-						if (!allowAddAssignmentUsers.contains(u))
-						{
-							returnResources.add(u);
-						}
+						returnResources.add(u);
 					}
 					catch (Exception e)
 					{
