Index: cluster/cluster-impl/impl/src/java/org/sakaiproject/cluster/impl/SakaiClusterService.java
===================================================================
--- cluster/cluster-impl/impl/src/java/org/sakaiproject/cluster/impl/SakaiClusterService.java	(revision 13762)
+++ cluster/cluster-impl/impl/src/java/org/sakaiproject/cluster/impl/SakaiClusterService.java	(working copy)
@@ -434,8 +434,10 @@
 
 						// find all the session ids of sessions that are open but are from closed servers
 						statement = "select SS.SESSION_ID " + "from SAKAI_SESSION SS "
-								+ "left join SAKAI_CLUSTER SC on SS.SESSION_SERVER = SC.SERVER_ID "
-								+ "where SS.SESSION_START = SS.SESSION_END " + "and SC.SERVER_ID is null";
+						    + "left join SAKAI_CLUSTER SC on SS.SESSION_SERVER = SC.SERVER_ID "
+						    + "where SS.SESSION_START = SS.SESSION_END " 
+						    + "AND SS.SESSION_START>=SYSDATE-7 "
+						    + "and SC.SERVER_ID is null";
 
 						List sessions = m_sqlService.dbRead(statement);
 
Index: event/event-impl/impl/src/java/org/sakaiproject/event/impl/UsageSessionServiceAdaptor.java
===================================================================
--- event/event-impl/impl/src/java/org/sakaiproject/event/impl/UsageSessionServiceAdaptor.java	(revision 13762)
+++ event/event-impl/impl/src/java/org/sakaiproject/event/impl/UsageSessionServiceAdaptor.java	(working copy)
@@ -1269,7 +1269,9 @@
 
 			// check the db
 			String statement = "select SESSION_ID,SESSION_SERVER,SESSION_USER,SESSION_IP,SESSION_USER_AGENT,SESSION_START,SESSION_END"
-					+ " from SAKAI_SESSION where SESSION_START = SESSION_END ORDER BY SESSION_SERVER ASC, SESSION_START ASC";
+			    + "  from SAKAI_SESSION where SESSION_START = SESSION_END" 
+			    + " AND SESSION_START >=SYSDATE-7"
+			    +"  ORDER BY SESSION_SERVER ASC, SESSION_START ASC";
 
 			List sessions = sqlService().dbRead(statement, null, new SqlReader()
 			{
