Index: webservices/axis/src/webapp/SakaiScript.jws
===================================================================
--- webservices/axis/src/webapp/SakaiScript.jws	(revision 34389)
+++ webservices/axis/src/webapp/SakaiScript.jws	(working copy)
@@ -233,6 +233,25 @@
 	return user.getEmail();
 }
 
+//getUserMyWorkspaceSiteId: return the site id of this users myworkspace site.
+public String getUserMyWorkspaceSiteId(String sessionid, String eid) throws AxisFault
+{
+	Session s = establishSession(sessionid);
+	
+	try {
+		User u = null;
+		String userid = UserDirectoryService.getUserByEid(eid).getId();
+		if (userid != null)
+		    return "~"+userid;
+		else
+		    return "";
+	}
+	catch (Exception e)
+	{
+	    return "";
+	}
+}
+
 public String getUserDisplayName( String sessionid ) throws AxisFault
 {
 	Session session = establishSession(sessionid);
