svn diff -r33453:33454 https://source.sakaiproject.org/svn
Index: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java
===================================================================
--- site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 33453)
+++ site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 33454)
@@ -4832,9 +4832,7 @@
 			isFutureTermSelected(state);
 		} else if (option.equalsIgnoreCase("cancel_edit")) {
 			// cancel
-			state.removeAttribute(STATE_TERM_SELECTED);
-			removeAddClassContext(state);
-			state.setAttribute(STATE_TEMPLATE_INDEX, "43");
+			doCancel(data);
 		} else if (option.equalsIgnoreCase("add")) {
 			isFutureTermSelected(state);
 			// continue
@@ -5787,11 +5785,13 @@
 				state.setAttribute(STATE_TEMPLATE_INDEX, "12");
 			}
 			removeEditToolState(state);
-		} else if (currentIndex.equals("37") || currentIndex.equals("44")) {
+		} else if (currentIndex.equals("37") || currentIndex.equals("44") || currentIndex.equals("53") || currentIndex.equals("36")) {
 			// cancel back to edit class view
+			state.removeAttribute(STATE_TERM_SELECTED);
+			removeAddClassContext(state);
 			state.setAttribute(STATE_TEMPLATE_INDEX, "43");
-			removeAddClassContext(state);
 		}
+		
 
 	} // doCancel
 
@@ -12233,7 +12233,14 @@
 				doBack(data);
 				return;
 			} else if ("cancel".equals(option)) {
-				doCancel_create(data);
+				if (getStateSite(state) == null) 
+				{
+					doCancel_create(data);// cancel from new site creation
+				}
+				else
+				{
+					doCancel(data);// cancel from site info editing
+				}
 				return;
 			} else if (option.equals("add")) {
 				addRequestedSection(state);
