svn diff -r33751:33752 https://source.sakaiproject.org/svn
Index: portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/SiteHandler.java
===================================================================
--- portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/SiteHandler.java	(revision 33751)
+++ portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/handlers/SiteHandler.java	(revision 33752)
@@ -71,9 +71,12 @@
 
 	private PortalSiteHelper siteHelper = new PortalSiteHelper();
 
+	private int configuredTabsToDisplay = 5;
+
 	public SiteHandler()
 	{
 		urlFragment = "site";
+        configuredTabsToDisplay  = ServerConfigurationService.getInt(Portal.CONFIG_DEFAULT_TABS, 5);
 	}
 
 	@Override
@@ -346,7 +349,7 @@
 				myWorkspaceSiteId = siteHelper.getSiteEffectiveId(myWorkspaceSite);
 			}
 
-			int tabsToDisplay = 4;
+            int tabsToDisplay = configuredTabsToDisplay;
 
 			if (!loggedIn)
 			{
Index: portal/portal-api/api/src/java/org/sakaiproject/portal/api/Portal.java
===================================================================
--- portal/portal-api/api/src/java/org/sakaiproject/portal/api/Portal.java	(revision 33751)
+++ portal/portal-api/api/src/java/org/sakaiproject/portal/api/Portal.java	(revision 33752)
@@ -86,6 +86,11 @@
 	 * Configuration option to enable/disable state reset on navigation change
 	 */
 	public static final String CONFIG_AUTO_RESET = "portal.experimental.auto.reset";
+	
+    /**
+	* Configuration option for default number of site tabs to display to users
+	*/
+	public static final String CONFIG_DEFAULT_TABS = "portal.default.tabs";
 
 	/**
 	 * Names of tool config/registration attributes that control the rendering
@@ -99,6 +104,7 @@
 
 	public static final String TOOLCONFIG_HELP_DOCUMENT_URL = "help.url";
 
+
 	/**
 	 * prepare the response and send it to the render engine
 	 * 
