svn diff -r32916:32917 https://source.sakaiproject.org/svn
Index: chat/chat-tool/tool/src/java/org/sakaiproject/chat2/tool/ChatTool.java
===================================================================
--- chat/chat-tool/tool/src/java/org/sakaiproject/chat2/tool/ChatTool.java	(revision 32916)
+++ chat/chat-tool/tool/src/java/org/sakaiproject/chat2/tool/ChatTool.java	(revision 32917)
@@ -32,6 +32,7 @@
 import javax.faces.context.ExternalContext;
 import javax.faces.context.FacesContext;
 import javax.faces.model.SelectItem;
+import javax.servlet.http.HttpServletRequest;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
@@ -50,6 +51,7 @@
 import org.sakaiproject.site.cover.SiteService;
 import org.sakaiproject.tool.api.Placement;
 import org.sakaiproject.tool.api.Session;
+import org.sakaiproject.tool.api.Tool;
 import org.sakaiproject.tool.api.ToolManager;
 import org.sakaiproject.tool.cover.SessionManager;
 import org.sakaiproject.user.api.User;
@@ -171,6 +173,7 @@
    /** the worksite the tool is in */
    private Site worksite = null;
    
+   private String toolContext = null;
    
    /*  error conditions */
    /** an error that could display on the select a chat room page */
@@ -214,6 +217,12 @@
       
       ExternalContext context = FacesContext.getCurrentInstance().getExternalContext();
       
+      HttpServletRequest req = (HttpServletRequest) context.getRequest();
+      req.setAttribute(Tool.NATIVE_URL, null); //signal to WrappedRequest that we want the Sakai managed
+      setToolContext(req.getContextPath());
+      req.setAttribute(Tool.NATIVE_URL, Tool.NATIVE_URL);
+      
+      
       try {
            context.redirect(url);
       }
@@ -801,6 +810,15 @@
     * @return the currentMessage
     */
    public DecoratedChatMessage getCurrentMessage() {
+      DecoratedChatMessage tmpCurrent = null;
+      if (currentMessage == null) {         
+         String messageId = (String)SessionManager.getCurrentToolSession().getAttribute("current_message");
+         if(messageId != null) {
+            ChatMessage message = getChatManager().getMessage(messageId);
+            tmpCurrent = new DecoratedChatMessage(this, message);
+            return tmpCurrent;
+         }
+      }
       return currentMessage;
    }
 
@@ -1248,6 +1266,14 @@
       }
       return toolBundle.getString(key);
    }
+
+   public String getToolContext() {
+      return toolContext;
+   }
+
+   public void setToolContext(String toolContext) {
+      this.toolContext = toolContext;
+   }
    
    
 }
Index: chat/chat-tool/tool/src/webapp/tools/sakai.chat.deleteMessage.xml
===================================================================
--- chat/chat-tool/tool/src/webapp/tools/sakai.chat.deleteMessage.xml	(revision 0)
+++ chat/chat-tool/tool/src/webapp/tools/sakai.chat.deleteMessage.xml	(revision 32917)
@@ -0,0 +1,10 @@
+<?xml version="1.0"?>
+
+<registration>
+
+	<tool id="sakai.chat.deleteMessage" title="Delete Chat Message" description="Chat Message Deletion Helper Tool">
+       <category name="sakai.helper" />
+
+	</tool>
+
+</registration>

Property changes on: chat/chat-tool/tool/src/webapp/tools/sakai.chat.deleteMessage.xml
___________________________________________________________________
Name: svn:eol-style
   + native

Index: chat/chat-tool/tool/src/webapp/WEB-INF/web.xml
===================================================================
--- chat/chat-tool/tool/src/webapp/WEB-INF/web.xml	(revision 32916)
+++ chat/chat-tool/tool/src/webapp/WEB-INF/web.xml	(revision 32917)
@@ -26,43 +26,19 @@
         <dispatcher>FORWARD</dispatcher>
         <dispatcher>INCLUDE</dispatcher>
     </filter-mapping>
+    
+    <filter-mapping>
+        <filter-name>sakai.request</filter-name>
+        <servlet-name>sakai.chat.deleteMessage</servlet-name>
+        <dispatcher>REQUEST</dispatcher>
+        <dispatcher>FORWARD</dispatcher>
+        <dispatcher>INCLUDE</dispatcher>
+    </filter-mapping>
 
     <listener>
-        <listener-class>org.sakaiproject.util.ContextLoaderListener</listener-class>
-    </listener>
-    <listener>
         <listener-class>org.sakaiproject.util.ToolListener</listener-class>
     </listener>
     
-	<!-- the vm handling servlet : used direct by name, not by URL path -->
-	<!--  <servlet>
-		<servlet-name>
-		   sakai.vm
-		</servlet-name>
-		<servlet-class>
-			org.sakaiproject.vm.VelocityServlet
-		</servlet-class>
-		<init-param>
-			<param-name> properties </param-name>
-			<param-value> velocity.properties </param-value>
-		</init-param>
-		<load-on-startup>1</load-on-startup>
-	</servlet>
-
-    <servlet>
-        <servlet-name>
-           sakai.chat
-        </servlet-name>
-        <servlet-class>
-            org.sakaiproject.chat.tool.ChatAction
-        </servlet-class>
-        <init-param>
-            <param-name>template</param-name>
-            <param-value>chat/chef_chat</param-value>            
-        </init-param>
-        <load-on-startup>1</load-on-startup>
-     </servlet>
-     -->
 	<!-- the faces handling xml -->
 	
     <context-param>
@@ -73,12 +49,8 @@
         <param-name>com.sun.faces.verifyObjects</param-name>
         <param-value>true</param-value>
     </context-param>
-
-    <context-param>
-       <param-name>contextConfigLocation</param-name>
-       <param-value>/WEB-INF/faces-config.xml</param-value>
-    </context-param>
-    <servlet>
+    
+   <servlet>
         <servlet-name>Faces Servlet</servlet-name>
         <servlet-class>javax.faces.webapp.FacesServlet</servlet-class>
         <load-on-startup> 2 </load-on-startup>
@@ -123,7 +95,26 @@
         </init-param>
       <load-on-startup>1</load-on-startup>
    </servlet>
+   
+     <servlet>
+      <servlet-name>sakai.chat.deleteMessage</servlet-name>
+      <servlet-class>org.sakaiproject.jsf.util.HelperAwareJsfTool</servlet-class>
+        <init-param>
+            <param-name>path</param-name>
+            <param-value>/jsp</param-value>
+        </init-param>
 
+        <init-param>
+            <param-name>default</param-name>
+            <!-- the path is inserted before and .jsp inserted after -->
+            <param-value>deleteMessageConfirm</param-value>
+        </init-param>
+         <init-param>
+            <param-name>default.last.view</param-name>
+            <param-value>false</param-value>
+        </init-param>
+      <load-on-startup>1</load-on-startup>
+   </servlet>
 
     <servlet-mapping>
         <servlet-name>Faces Servlet</servlet-name>
Index: chat/chat-tool/tool/src/webapp/js/chatscript.js
===================================================================
--- chat/chat-tool/tool/src/webapp/js/chatscript.js	(revision 32916)
+++ chat/chat-tool/tool/src/webapp/js/chatscript.js	(revision 32917)
@@ -59,9 +59,9 @@
 	{
 		newComponentId = $(transcript).children("li").size();
 		var builtId = "topForm:chatList:" + newComponentId + ":deleteMessage";
-		deleteUrl = "document.forms['topForm']['topForm:_idcl'].value='" + builtId + "'; document.forms['topForm'].submit(); return false;";
+		var tmpdeleteUrl = deleteUrl + msgId;
 		deleteHtml = 
-			" <a id=\"" + builtId + "\" href=\"#\" onclick=\"" + deleteUrl + "\" title=\"" + deleteMsg + "\" >" +
+			" <a id=\"" + builtId + "\" href=\"#\" onclick=\"location.href='" + tmpdeleteUrl + "'\" title=\"" + deleteMsg + "\" >" +
 			"<img src=\"/library/image/sakai/delete.gif\" border=\"0\" alt=\"" + deleteMsg + "\" /></a>";
 	}
 
@@ -75,9 +75,6 @@
 	var objDiv = document.getElementById("Monitor");
    objDiv.scrollTop = objDiv.scrollHeight;
 
-	//Force a refresh of the content (doesn't actually reload the page)
-   $.ajax({type: "GET", url: location.href, data: ""});
-
 }
 
 function escapeHTML(text) {
Index: chat/chat-tool/tool/src/webapp/jsp/roomMonitor.jspf
===================================================================
--- chat/chat-tool/tool/src/webapp/jsp/roomMonitor.jspf	(revision 32916)
+++ chat/chat-tool/tool/src/webapp/jsp/roomMonitor.jspf	(revision 32917)
@@ -10,6 +10,7 @@
 
 // set for the chatscript.js
 deleteMsg = "<h:outputText value="#{msgs['list.del']}" />";
+deleteUrl = "<h:outputText value="#{ChatTool.toolContext}/sakai.chat.deleteMessage.helper/deleteMessageConfirm?session.current_message=" />";
 </script>
 
 	<sakai:script contextBase="/sakai-chat-tool" path="/js/chatscript.js"/>
