svn diff -r33500:33501 https://source.sakaiproject.org/svn
Index: assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java
===================================================================
--- assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java	(revision 33500)
+++ assignment/assignment-tool/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java	(revision 33501)
@@ -2798,7 +2798,6 @@
 		state.setAttribute(STATE_MODE, MODE_LIST_ASSIGNMENTS);
 		state.setAttribute(SORTED_BY, SORTED_BY_DEFAULT);
 		state.setAttribute(SORTED_ASC, Boolean.TRUE.toString());
-		state.setAttribute(SORTED_ASC, Boolean.FALSE.toString());
 
 	} // doList_assignments
 
@@ -4313,7 +4312,7 @@
 	 */
 	private void setDefaultSort(SessionState state) {
 		state.setAttribute(SORTED_BY, SORTED_BY_DEFAULT);
-		state.setAttribute(SORTED_ASC, Boolean.FALSE.toString());
+		state.setAttribute(SORTED_ASC, Boolean.TRUE.toString());
 	}
 
 	/**
