svn diff -r31418:31419 https://source.sakaiproject.org/svn
Index: podcasts/podcasts-impl/impl/src/java/org/sakaiproject/component/app/podcasts/PodcastServiceImpl.java
===================================================================
--- podcasts/podcasts-impl/impl/src/java/org/sakaiproject/component/app/podcasts/PodcastServiceImpl.java	(revision 31418)
+++ podcasts/podcasts-impl/impl/src/java/org/sakaiproject/component/app/podcasts/PodcastServiceImpl.java	(revision 31419)
@@ -43,6 +43,7 @@
 import org.sakaiproject.content.api.ContentHostingService;
 import org.sakaiproject.content.api.ContentResource;
 import org.sakaiproject.content.api.ContentResourceEdit;
+import org.sakaiproject.content.api.GroupAwareEntity.AccessMode;
 import org.sakaiproject.entity.api.Entity;
 import org.sakaiproject.entity.api.EntityPropertyNotDefinedException;
 import org.sakaiproject.entity.api.EntityPropertyTypeException;
@@ -320,7 +321,8 @@
 	}
 
 	/**
-	 * Remove on file resources from list of potential podcasts
+	 * Remove non-file resources from list of potential podcasts and
+	 * files restricted to groups user not part of
 	 * 
 	 * @param resourcesList
 	 *           The list of potential podcasts
@@ -343,9 +345,11 @@
 
 				 aResource = (ContentResource) podcastIter.next();
 				
-				if (aResource.isResource() /*&& aResource.getProperties().getProperty(ResourceProperties.PROP_ORIGINAL_FILENAME) != null */) {
-					filteredResources.add(aResource);
-				
+				if (aResource.isResource()) {
+					if (aResource.getAccess().equals(AccessMode.GROUPED)) {
+						
+					}
+					filteredResources.add(aResource);				
 				}
 			} 
 			catch (ClassCastException e) {
@@ -432,9 +436,13 @@
 				// Does not exist, so try to create it
 				podcastsCollection = siteCollection + COLLECTION_PODCASTS + Entity.SEPARATOR;
 				
-				createPodcastsFolder(podcastsCollection, siteId);
-				
-				return podcastsCollection;
+				if (canUpdateSite()) {
+					createPodcastsFolder(podcastsCollection, siteId);
+					return podcastsCollection;
+				}
+				else {
+					return null;
+				}
 
 			}
 			catch (PermissionException e1) {
@@ -499,6 +507,9 @@
 			// not have DISPLAY_DATE property
 			final ContentCollection collectionEdit = getContentCollection(siteId);
 
+			// TODO: check if folder is restricted to group access
+			//    and if so, does this user have access?
+			
 			resourcesList = collectionEdit.getMemberResources();
 
 			// remove non-file resources from collection
@@ -517,7 +528,12 @@
 		} 
 		catch (IdUnusedException ex) {
 				// Does not exist, attempt to create it
+			if (canUpdateSite()) {
 				createPodcastsFolder(podcastsCollection, siteId);
+			}
+			else {
+				return new ArrayList();
+			}
 		}
 
 		return resourcesList;
@@ -1143,7 +1159,6 @@
 
 	/**
 	 * Checks if podcast feed title and description exists and if not, add them
-	 * TODO: move to BasicPodfeedService
 	 * 
 	 * @param podcastsCollection
 	 * 			The id for the podcasts collection
