Index: msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java
===================================================================
--- msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java	(revision 36827)
+++ msgcntr/messageforums-app/src/java/org/sakaiproject/tool/messageforums/PrivateMessagesTool.java	(working copy)
@@ -967,12 +967,16 @@
   {
     LOG.debug("processActionHome()");
     msgNavMode = "privateMessages";
+    if (searchPvtMsgs != null)
+     	searchPvtMsgs.clear();
     return  MAIN_PG;
   }  
   public String processActionPrivateMessages()
   {
     LOG.debug("processActionPrivateMessages()");                    
     msgNavMode = "privateMessages";            
+    if (searchPvtMsgs != null) 
+   	searchPvtMsgs.clear();
     return  MESSAGE_HOME_PG;
   }        
   public String processDisplayForum()
@@ -1643,6 +1647,11 @@
       //set prev/next message
       setPrevNextMessageDetails();
       
+      if (searchPvtMsgs != null)
+      {
+    	  searchPvtMsgs.clear();
+    	  return DISPLAY_MESSAGES_PG;
+      }
     }
     return null;
   }
@@ -1674,7 +1683,14 @@
       setPrevNextTopicDetails(msgNavMode);
       //set prev/next message
       setPrevNextMessageDetails();
+      
+      if (searchPvtMsgs != null)
+      {
+    	  searchPvtMsgs.clear();
+    	  return DISPLAY_MESSAGES_PG;
+      }
     }
+
     return null;
   }
 /////////////////////////////////////     DISPLAY NEXT/PREVIOUS TOPIC     //////////////////////////////////
@@ -3157,7 +3173,8 @@
     }
 
 
-    public String getAuthorString() {
+    public String getAuthorString() 
+    {
        String authorString = getUserId();
        
        try
@@ -3173,12 +3190,17 @@
        return authorString;
     }
     
-    public String getPlacementId() {
+    public String getPlacementId() 
+    {
        return Validator.escapeJavascript("Main" + ToolManager.getCurrentPlacement().getId());
     }
 
+    public boolean isSearchPvtMsgsEmpty()
+    {
+    	return searchPvtMsgs == null || searchPvtMsgs.isEmpty();
+    }
 
-	public void setMsgNavMode(String msgNavMode) {
+    public void setMsgNavMode(String msgNavMode) {
 		this.msgNavMode = msgNavMode;
 	}	
 	
Index: msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsgEx.jsp
===================================================================
--- msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsgEx.jsp	(revision 36827)
+++ msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsgEx.jsp	(working copy)
@@ -15,14 +15,7 @@
        		<sakai:script contextBase="/sakai-messageforums-tool" path="/js/sak-10625.js"/>
 			<sakai:script contextBase="/sakai-messageforums-tool" path="/js/forum.js"/>		
 			
-			<div class="breadCrumb">
-				<h3>
-				<h:commandLink action="#{PrivateMessagesTool.processActionHome}" value="#{msgs.cdfm_message_forums}" title=" #{msgs.cdfm_message_forums}"/> /
-	  		  <h:commandLink action="#{PrivateMessagesTool.processActionPrivateMessages}" value="#{msgs.cdfm_message_pvtarea}" title=" #{msgs.cdfm_message_forums}"/> /
-				<h:commandLink action="#{PrivateMessagesTool.processDisplayForum}" value="#{PrivateMessagesTool.msgNavMode}" title=" #{msgs.cdfm_message_forums}"/> /
-				<h:outputText value="#{msgs.pvt_search}" />
-				</h3>
-		  </div>
+			<%@include file="topNav.jsp" %>
 
 			<h:messages styleClass="alertMessage" id="errorMessages" />
 			<%@include file="msgHeader.jsp"%>
Index: msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsg.jsp
===================================================================
--- msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsg.jsp	(revision 36827)
+++ msgcntr/messageforums-app/src/webapp/jsp/privateMsg/pvtMsg.jsp	(working copy)
@@ -37,33 +37,7 @@
 			<sakai:script contextBase="/sakai-messageforums-tool" path="/js/forum.js"/>		
 			
 			<%--<sakai:tool_bar_message value="#{msgs.pvt_pvtmsg}- #{PrivateMessagesTool.msgNavMode}" /> --%>
-			<h:panelGrid columns="2" summary="layout" width="100%" styleClass="navPanel">
-        <h:panelGroup>
-          	<f:verbatim><div class="breadCrumb specialLink"><h3></f:verbatim>
-				  <h:panelGroup rendered="#{PrivateMessagesTool.messagesandForums}" >
-				  	<h:commandLink action="#{PrivateMessagesTool.processActionHome}" value="#{msgs.cdfm_message_forums}" title="#{msgs.cdfm_message_forums}"/>
-				  	<f:verbatim><h:outputText value=" / " /></f:verbatim>
-				  </h:panelGroup>
-	  		      <h:commandLink action="#{PrivateMessagesTool.processActionPrivateMessages}" value="#{msgs.pvt_message_nav}" title=" #{msgs.cdfm_message_forums}"/>
-	              <f:verbatim><h:outputText value=" " /><h:outputText value=" / " /><h:outputText value=" " /></f:verbatim>
-	 		      <h:outputText value="#{PrivateMessagesTool.msgNavMode}"/>
-			    <f:verbatim></h3></div></f:verbatim>
-        </h:panelGroup>
-        <h:panelGroup styleClass="itemNav specialLink">
-		<%-- gsilver:huh? renders anyway - because it is looking at topics instead of at folders?--%>
-				 <h:commandLink action="#{PrivateMessagesTool.processDisplayPreviousTopic}" value="#{msgs.pvt_prev_folder}"  
-				                rendered="#{PrivateMessagesTool.selectedTopic.hasPreviousTopic}" title=" #{msgs.pvt_prev_folder}">
-	  			   <f:param value="#{PrivateMessagesTool.selectedTopic.previousTopicTitle}" name="previousTopicTitle"/>
-	  		   </h:commandLink>
-				<h:outputText 	value="#{msgs.pvt_prev_folder}"  rendered="#{!PrivateMessagesTool.selectedTopic.hasPreviousTopic}" />
-				  <f:verbatim><h:outputText value=" | " /></f:verbatim>
-	  		   <h:commandLink action="#{PrivateMessagesTool.processDisplayNextTopic}" value="#{msgs.pvt_next_folder}" 
-	  		                  rendered="#{PrivateMessagesTool.selectedTopic.hasNextTopic}" title=" #{msgs.pvt_next_folder}">
-	  			  <f:param value="#{PrivateMessagesTool.selectedTopic.nextTopicTitle}" name="nextTopicTitle"/>
-	  		   </h:commandLink>
-			   				<h:outputText 	 value="#{msgs.pvt_next_folder}"  rendered="#{!PrivateMessagesTool.selectedTopic.hasNextTopic}" />
-        </h:panelGroup>
-      </h:panelGrid>
+			<%@include file="topNav.jsp" %>
  
  			<h:messages styleClass="alertMessage" id="errorMessages" /> 
   	
