svn diff -r34644:34645 https://source.sakaiproject.org/svn
Index: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java
===================================================================
--- content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java	(revision 34644)
+++ content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java	(revision 34645)
@@ -455,6 +455,9 @@
 		String typeId = pipe.getAction().getTypeId();
 		String mimetype = pipe.getMimeType();
 		
+		ListItem item = new ListItem(pipe.getContentEntity());
+		context.put("item", item);
+		
 		// context.put("inDropbox", ContentHostingService.isInDropbox(pipe.getContentEntity().getId()));
 		ResourceTypeRegistry registry = (ResourceTypeRegistry) ComponentManager.get(ResourceTypeRegistry.class);
 		if(registry != null)
Index: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ListItem.java
===================================================================
--- content/content-tool/tool/src/java/org/sakaiproject/content/tool/ListItem.java	(revision 34644)
+++ content/content-tool/tool/src/java/org/sakaiproject/content/tool/ListItem.java	(revision 34645)
@@ -846,7 +846,7 @@
 
 		}
 		
-		if(this.id != null)
+		if(this.id != null && ! this.id.trim().equals(""))
 		{
 			this.isDropbox = contentService.isInDropbox(id);
 		}
Index: content/content-tool/tool/src/webapp/vm/content/sakai_resources_cwiz_finish.vm
===================================================================
--- content/content-tool/tool/src/webapp/vm/content/sakai_resources_cwiz_finish.vm	(revision 34644)
+++ content/content-tool/tool/src/webapp/vm/content/sakai_resources_cwiz_finish.vm	(revision 34645)
@@ -28,7 +28,7 @@
 			<hr class="itemSeparator" />
 		</div>		
 	
-		#if (!$dropboxMode)
+		#if(! $item.isDropbox())
 			#if ($type.hasNotificationDialog())
 				#if ($notification)
 					<p class="shorttext">
Index: content/content-tool/tool/src/webapp/vm/resources/sakai_create_urls.vm
===================================================================
--- content/content-tool/tool/src/webapp/vm/resources/sakai_create_urls.vm	(revision 34644)
+++ content/content-tool/tool/src/webapp/vm/resources/sakai_create_urls.vm	(revision 34645)
@@ -58,7 +58,7 @@
 		<p class="act">
 			<a href="#" onclick="javascript:addFileInput();">$tlang.getString("label.addurl")</a>
 		</p>
-		#if(!$dropboxMode)
+		#if(!$model.isDropbox())
 			#if ($model.resourceTypeDef.hasNotificationDialog())
 				<p class="shorttext">
 					<label for="notify">
Index: content/content-tool/tool/src/webapp/vm/resources/sakai_create_uploads.vm
===================================================================
--- content/content-tool/tool/src/webapp/vm/resources/sakai_create_uploads.vm	(revision 34644)
+++ content/content-tool/tool/src/webapp/vm/resources/sakai_create_uploads.vm	(revision 34645)
@@ -57,7 +57,7 @@
 		<p class="act">
 			<a href="#" onclick="javascript:addFileInput();">$tlang.getString("label.addfile")</a>
 		</p>
-		#if(!$dropboxMode)
+		#if(!$model.isDropbox())
 			#if ($model.resourceTypeDef.hasNotificationDialog())
 				<p class="shorttext">
 					<label for="notify">
Index: content/content-tool/tool/src/webapp/vm/resources/sakai_replace_file.vm
===================================================================
--- content/content-tool/tool/src/webapp/vm/resources/sakai_replace_file.vm	(revision 34644)
+++ content/content-tool/tool/src/webapp/vm/resources/sakai_replace_file.vm	(revision 34645)
@@ -39,17 +39,19 @@
 				</td>
 			</tr>
 		</table>
-		#if ($item.resourceTypeDef.hasNotificationDialog())
-			<p class="shorttext">
-				<label for="notify">
-					$tlang.getString("gen.email1")
-				</label>	
-				<select name="notify" id="notify">
-					<option value="r">$tlang.getString("gen.email2")</option>
-					<option value="o">$tlang.getString("gen.email3")</option>
-					<option value="n" selected="selected">$tlang.getString("gen.email4")</option>
-				</select>
-			</p>	
+		#if(! $item.isDropbox())
+			#if ($item.resourceTypeDef.hasNotificationDialog())
+				<p class="shorttext">
+					<label for="notify">
+						$tlang.getString("gen.email1")
+					</label>	
+					<select name="notify" id="notify">
+						<option value="r">$tlang.getString("gen.email2")</option>
+						<option value="o">$tlang.getString("gen.email3")</option>
+						<option value="n" selected="selected">$tlang.getString("gen.email4")</option>
+					</select>
+				</p>	
+			#end
 		#end
 		
 		<p class="act">
Index: content/content-tool/tool/src/webapp/vm/resources/sakai_revise_url.vm
===================================================================
--- content/content-tool/tool/src/webapp/vm/resources/sakai_revise_url.vm	(revision 34644)
+++ content/content-tool/tool/src/webapp/vm/resources/sakai_revise_url.vm	(revision 34645)
@@ -1,4 +1,4 @@
-<!-- resources/sakai_edit_text.vm, use with org.sakaiproject.tool.content.ResourcesHelperAction.java -->
+<!-- resources/sakai_revise_url.vm, use with org.sakaiproject.tool.content.ResourcesHelperAction.java -->
 <div class="portletBody specialLink">
 	<h3>
 		$tlang.getString("revise.url")
@@ -16,17 +16,19 @@
 			<label for="content" class="block">$tlang.getString("label.url")</label>
 			<input type="text" name="content" id="content" size="60" value="$!validator.escapeHtml("$!pipe.contentstring")" />
 		</p>
-		#if ($hasNotificationDialog)
-			<p class="longtext" >
-				<label for="notify" class="block">
-					$tlang.getString("gen.email1")
-				</label>	
-				<select name="notify" id="notify">
-					<option value="r">$tlang.getString("gen.email2")</option>
-					<option value="o">$tlang.getString("gen.email3")</option>
-					<option value="n" selected="selected">$tlang.getString("gen.email4")</option>
-				</select>
-			</p>	
+		#if(! $item.isDropbox())
+			#if ($hasNotificationDialog)
+				<p class="shorttext">
+					<label for="notify">
+						$tlang.getString("gen.email1")
+					</label>	
+					<select name="notify" id="notify">
+						<option value="r">$tlang.getString("gen.email2")</option>
+						<option value="o">$tlang.getString("gen.email3")</option>
+						<option value="n" selected="selected">$tlang.getString("gen.email4")</option>
+					</select>
+				</p>	
+			#end
 		#end
 		<p class="act">
 			<input type="button" class="active" name="savechanges" id="saveChanges" onclick="javascript:showNotif('submitnotif','saveChanges','addContentForm');document.getElementById('addContentForm').action='#toolLink("ResourcesHelperAction" "doContinue")';submitform('addContentForm');" value="$tlang.getString("label.continue")"  accesskey="s"/>
Index: content/content-tool/tool/src/webapp/vm/resources/sakai_revise_html.vm
===================================================================
--- content/content-tool/tool/src/webapp/vm/resources/sakai_revise_html.vm	(revision 34644)
+++ content/content-tool/tool/src/webapp/vm/resources/sakai_revise_html.vm	(revision 34645)
@@ -22,17 +22,19 @@
 				</tr>
 			</table>
 		</div>
-		#if ($hasNotificationDialog)
-			<p class="shorttext">
-				<label for="notify">
-					$tlang.getString("gen.email1")
-				</label>	
-				<select name="notify" id="notify">
-					<option value="r">$tlang.getString("gen.email2")</option>
-					<option value="o">$tlang.getString("gen.email3")</option>
-					<option value="n" selected="selected">$tlang.getString("gen.email4")</option>
-				</select>
-			</p>	
+		#if(! $model.isDropbox())
+			#if ($hasNotificationDialog)
+				<p class="shorttext">
+					<label for="notify">
+						$tlang.getString("gen.email1")
+					</label>	
+					<select name="notify" id="notify">
+						<option value="r">$tlang.getString("gen.email2")</option>
+						<option value="o">$tlang.getString("gen.email3")</option>
+						<option value="n" selected="selected">$tlang.getString("gen.email4")</option>
+					</select>
+				</p>	
+			#end
 		#end
 		<p class="act">
 			<input type="button" class="active" name="savechanges" id="saveChanges" onclick="javascript:showNotif('submitnotif','saveChanges','addContentForm');document.getElementById('addContentForm').action='#toolLink("ResourcesHelperAction" "doContinue")';submitform('addContentForm');" value="$tlang.getString("label.continue")" accesskey="s" />
Index: content/content-tool/tool/src/webapp/vm/resources/sakai_revise_text.vm
===================================================================
--- content/content-tool/tool/src/webapp/vm/resources/sakai_revise_text.vm	(revision 34644)
+++ content/content-tool/tool/src/webapp/vm/resources/sakai_revise_text.vm	(revision 34645)
@@ -15,17 +15,19 @@
 		#end
 			<textarea name="content" id="content" rows="10" cols="60" wrap="virtual">$!validator.escapeHtmlTextarea($!{pipe.contentstring})</textarea>
 		</p>
-		#if ($hasNotificationDialog)
-			<p class="shorttext">
-				<label for="notify">
-					$tlang.getString("gen.email1")
-				</label>	
-				<select name="notify" id="notify">
-					<option value="r">$tlang.getString("gen.email2")</option>
-					<option value="o">$tlang.getString("gen.email3")</option>
-					<option value="n" selected="selected">$tlang.getString("gen.email4")</option>
-				</select>
-			</p>	
+		#if(! $item.isDropbox())
+			#if ($hasNotificationDialog)
+				<p class="shorttext">
+					<label for="notify">
+						$tlang.getString("gen.email1")
+					</label>	
+					<select name="notify" id="notify">
+						<option value="r">$tlang.getString("gen.email2")</option>
+						<option value="o">$tlang.getString("gen.email3")</option>
+						<option value="n" selected="selected">$tlang.getString("gen.email4")</option>
+					</select>
+				</p>	
+			#end
 		#end
 		<p class="act">
 			<input type="button" class="active" name="savechanges" id="saveChanges" onclick="javascript:showNotif('submitnotif','saveChanges','addContentForm');document.getElementById('addContentForm').action='#toolLink("ResourcesHelperAction" "doContinue")';submitform('addContentForm');" value="$tlang.getString("label.continue")" accesskey="s" />
