svn diff -r41496:41497 https://source.sakaiproject.org/svn
Index: announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/SiteEmailNotificationAnnc.java
===================================================================
--- announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/SiteEmailNotificationAnnc.java	(revision 41496)
+++ announcement/announcement-impl/impl/src/java/org/sakaiproject/announcement/impl/SiteEmailNotificationAnnc.java	(revision 41497)
@@ -54,7 +54,9 @@
 import org.sakaiproject.time.api.Time;
 import org.sakaiproject.time.cover.TimeService;
 import org.sakaiproject.util.EmailNotification;
+import org.sakaiproject.util.FormattedText;
 import org.sakaiproject.util.SiteEmailNotification;
+import org.sakaiproject.util.StringUtil;
 import org.sakaiproject.component.cover.ComponentManager;
 import org.sakaiproject.util.ResourceLoader;
 
@@ -397,7 +399,9 @@
 	@Override
 	protected String plainTextContent() {
 		// TODO replace this with a reasonable plain text equivalent
-		return htmlContent();
+		String content = htmlContent();
+		content = FormattedText.convertFormattedTextToPlaintext(content);
+		return content;
 	}
 
 }
