svn diff -r19792:19793 https://source.sakaiproject.org/svn
Index: gradebook/app/project.xml
===================================================================
--- gradebook/app/project.xml	(revision 19792)
+++ gradebook/app/project.xml	(revision 19793)
@@ -271,7 +271,7 @@
 		<dependency>
 			<groupId>commons-fileupload</groupId>
 			<artifactId>commons-fileupload</artifactId>
-			<version>1.1</version>
+			<version>1.1.1</version>
 			<properties>
 				<war.bundle>true</war.bundle>
 			</properties>
Index: textarea/.classpath
===================================================================
--- textarea/.classpath	(revision 19792)
+++ textarea/.classpath	(revision 19793)
@@ -4,7 +4,7 @@
 	<classpathentry sourcepath="JRE_SRC" kind="var" path="JRE_LIB"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-logging/jars/commons-logging-1.0.4.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/servletapi/jars/servletapi-2.4.jar"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/entity-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/content-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/event-api"/>
Index: textarea/FCKeditor/connector/project.xml
===================================================================
--- textarea/FCKeditor/connector/project.xml	(revision 19792)
+++ textarea/FCKeditor/connector/project.xml	(revision 19793)
@@ -53,7 +53,7 @@
 		<dependency>
 			<groupId>commons-fileupload</groupId>
 			<artifactId>commons-fileupload</artifactId>
-			<version>1.0</version>
+			<version>1.1.1</version>
 		</dependency>
 
 		<dependency>
Index: sam/samigo-app/project.xml
===================================================================
--- sam/samigo-app/project.xml	(revision 19792)
+++ sam/samigo-app/project.xml	(revision 19793)
@@ -401,7 +401,7 @@
     <dependency>
       <groupId>commons-fileupload</groupId>
       <artifactId>commons-fileupload</artifactId>
-      <version>1.0</version>
+      <version>1.1.1</version>
       <properties>
         <war.bundle>standalone</war.bundle>
       </properties>
Index: sam/.classpath
===================================================================
--- sam/.classpath	(revision 19792)
+++ sam/.classpath	(revision 19793)
@@ -22,7 +22,7 @@
 	<classpathentry kind="var" path="MAVEN_REPO/commons-dbcp/jars/commons-dbcp-1.2.1.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-digester/jars/commons-digester-1.6.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-discovery/jars/commons-discovery-0.2.jar"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-id/jars/commons-id-0.1-dev.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-jxpath/jars/commons-jxpath-1.1.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-lang/jars/commons-lang-2.1.jar"/>
Index: sam/samigo-services/project.xml
===================================================================
--- sam/samigo-services/project.xml	(revision 19792)
+++ sam/samigo-services/project.xml	(revision 19793)
@@ -219,7 +219,7 @@
     <dependency>
       <groupId>commons-fileupload</groupId>
       <artifactId>commons-fileupload</artifactId>
-      <version>1.0</version>
+      <version>1.1.1</version>
     </dependency>
     <dependency>
       <groupId>commons-id</groupId>
Index: sam/samigo-hibernate/project.xml
===================================================================
--- sam/samigo-hibernate/project.xml	(revision 19792)
+++ sam/samigo-hibernate/project.xml	(revision 19793)
@@ -208,7 +208,7 @@
     <dependency>
       <groupId>commons-fileupload</groupId>
       <artifactId>commons-fileupload</artifactId>
-      <version>1.0</version>
+      <version>1.1.1</version>
     </dependency>
     <dependency>
       <groupId>commons-id</groupId>
Index: metaobj/metaobj-impl/.classpath
===================================================================
--- metaobj/metaobj-impl/.classpath	(revision 19792)
+++ metaobj/metaobj-impl/.classpath	(revision 19793)
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="api-impl/src/java"/>
-	<classpathentry exported="true" kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry exported="true" kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry exported="true" sourcepath="JRE_SRC" kind="var" path="JRE_LIB"/>
 	<classpathentry kind="var" path="MAVEN_REPO/org.springframework/jars/spring-1.2.8.jar"/>
 	<classpathentry exported="true" kind="var" path="MAVEN_REPO/velocity/jars/velocity-1.3.1.jar"/>
Index: metaobj/metaobj-api/.classpath
===================================================================
--- metaobj/metaobj-api/.classpath	(revision 19792)
+++ metaobj/metaobj-api/.classpath	(revision 19793)
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <classpath>
 	<classpathentry kind="src" path="api/src/java"/>
-	<classpathentry exported="true" kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry exported="true" kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry exported="true" sourcepath="JRE_SRC" kind="var" path="JRE_LIB"/>
 	<classpathentry exported="true" kind="var" path="MAVEN_REPO/servletapi/jars/servletapi-2.4.jar"/>
 	<classpathentry exported="true" kind="var" path="MAVEN_REPO/jsf/jars/jsf-impl-1.1.01.jar"/>
Index: jsf/.classpath
===================================================================
--- jsf/.classpath	(revision 19792)
+++ jsf/.classpath	(revision 19793)
@@ -7,7 +7,7 @@
 	<classpathentry excluding="**/.DS_Store" kind="src" path="widgets/src/java"/>
 	<classpathentry excluding="**/.DS_Store" kind="src" path="widgets-sun/src/java"/>
 	<classpathentry excluding="**/JSFDepends.java" kind="src" path="widgets-myfaces/src/java"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry sourcepath="JRE_SRC" kind="var" path="JRE_LIB"/>
 	<classpathentry kind="var" path="MAVEN_REPO/servletapi/jars/servletapi-2.4.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-impl-1.1.01.jar"/>
Index: jsf/widgets/project.xml
===================================================================
--- jsf/widgets/project.xml	(revision 19792)
+++ jsf/widgets/project.xml	(revision 19793)
@@ -94,7 +94,7 @@
      <dependency>
       <groupId>commons-fileupload</groupId>
       <artifactId>commons-fileupload</artifactId>
-      <version>1.0</version>
+      <version>1.1.1</version>
       <url>http://jakarta.apache.org/commons/fileupload/</url>
     </dependency>
 
Index: jsf/example/project.xml
===================================================================
--- jsf/example/project.xml	(revision 19792)
+++ jsf/example/project.xml	(revision 19793)
@@ -190,7 +190,7 @@
 		<dependency>
 			<groupId>commons-fileupload</groupId>
 			<artifactId>commons-fileupload</artifactId>
-			<version>1.0</version>
+			<version>1.1.1</version>
 		</dependency>
 
 		<dependency>
Index: msgcntr/.classpath
===================================================================
--- msgcntr/.classpath	(revision 19792)
+++ msgcntr/.classpath	(revision 19793)
@@ -8,7 +8,7 @@
 	<classpathentry kind="src" path="messageforums-app/src/java"/>
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-impl-1.1.01.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-api-1.1.01.jar"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
        <classpathentry kind="var" path="MAVEN_REPO/xerces/jars/xercesImpl-2.6.2.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-logging/jars/commons-logging-1.0.4.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-lang/jars/commons-lang-2.0.jar"/>
Index: podcasts/.classpath
===================================================================
--- podcasts/.classpath	(revision 19792)
+++ podcasts/.classpath	(revision 19793)
@@ -8,7 +8,7 @@
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-api-1.1.01.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-impl-1.1.01.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/jspapi/jars/jsp-api-2.0.jar"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/content-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/site-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/tool-api"/>
Index: syllabus/.classpath
===================================================================
--- syllabus/.classpath	(revision 19792)
+++ syllabus/.classpath	(revision 19793)
@@ -7,7 +7,7 @@
 	<classpathentry kind="src" path="syllabus-app/src/java"/>
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-impl-1.1.01.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-api-1.1.01.jar"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/org.hibernate/jars/hibernate-3.1.3.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/jsf"/>
 	<classpathentry kind="src" path="/entity-api"/>
Index: velocity/.classpath
===================================================================
--- velocity/.classpath	(revision 19792)
+++ velocity/.classpath	(revision 19793)
@@ -6,7 +6,7 @@
 	<classpathentry kind="var" path="MAVEN_REPO/commons-logging/jars/commons-logging-1.0.4.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/servletapi/jars/servletapi-2.4.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/velocity/jars/velocity-1.3.1.jar"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/event-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/entity-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/courier-api"/>
Index: velocity/tool/project.xml
===================================================================
--- velocity/tool/project.xml	(revision 19792)
+++ velocity/tool/project.xml	(revision 19793)
@@ -95,7 +95,7 @@
 		<dependency>
 			<groupId>commons-fileupload</groupId>
 			<artifactId>commons-fileupload</artifactId>
-			<version>1.0</version>
+			<version>1.1.1</version>
 		</dependency>
 
 		<dependency>
Index: access/access-impl/.classpath
===================================================================
--- access/access-impl/.classpath	(revision 19792)
+++ access/access-impl/.classpath	(revision 19793)
@@ -12,7 +12,7 @@
 	<classpathentry combineaccessrules="false" kind="src" path="/content-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/user-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/event-api"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/util-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/util-util"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/component-api"/>
Index: access/access-impl/impl/project.xml
===================================================================
--- access/access-impl/impl/project.xml	(revision 19792)
+++ access/access-impl/impl/project.xml	(revision 19793)
@@ -59,7 +59,7 @@
 		<dependency>
 			<groupId>commons-fileupload</groupId>
 			<artifactId>commons-fileupload</artifactId>
-			<version>1.0</version>
+			<version>1.1.1</version>
 		</dependency>
 
 		<dependency>
Index: postem/.classpath
===================================================================
--- postem/.classpath	(revision 19792)
+++ postem/.classpath	(revision 19793)
@@ -8,7 +8,7 @@
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-api-1.1.01.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/jsf/jars/jsf-impl-1.1.01.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/taglibs/jars/standard-1.1.2.jar"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry sourcepath="JAVA_DIR/myfaces-1.0.9-src" kind="var" path="MAVEN_REPO/myfaces/jars/myfaces-1.0.9.jar"/>
 	<classpathentry kind="var" path="MAVEN_REPO/commons-math/jars/commons-math-1.0.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/user-api"/>

Property changes on: postem/postem-help
___________________________________________________________________
Name: svn:ignore
   + target


Index: tool/tool-util/.classpath
===================================================================
--- tool/tool-util/.classpath	(revision 19792)
+++ tool/tool-util/.classpath	(revision 19793)
@@ -9,7 +9,7 @@
 	<classpathentry kind="var" path="MAVEN_REPO/concurrent/jars/concurrent-1.3.4.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/authz-api"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/tool-api"/>
-	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.0.jar"/>
+	<classpathentry kind="var" path="MAVEN_REPO/commons-fileupload/jars/commons-fileupload-1.1.1.jar"/>
 	<classpathentry combineaccessrules="false" kind="src" path="/util-api"/>
 	<classpathentry kind="output" path="bin"/>
 </classpath>
Index: tool/tool-util/servlet/project.xml
===================================================================
--- tool/tool-util/servlet/project.xml	(revision 19792)
+++ tool/tool-util/servlet/project.xml	(revision 19793)
@@ -49,7 +49,7 @@
 		<dependency>
 			<groupId>commons-fileupload</groupId>
 			<artifactId>commons-fileupload</artifactId>
-			<version>1.0</version>
+			<version>1.1.1</version>
 			<url>http://jakarta.apache.org/commons/fileupload/</url>
 			<properties>
 				<!-- deploy dependency jar to "shared", "common" or "server": use <war.bundle>true</war.bundle> to include in war -->
@@ -57,6 +57,18 @@
 			</properties>
 		</dependency>
 
+ 		<!-- commons-fileupload is dependent on this artifact -->
+ 		<dependency>
+			<groupId>commons-io</groupId>
+			<artifactId>commons-io</artifactId>
+			<version>1.2</version>
+			<url>http://jakarta.apache.org/commons/io/</url>
+			<properties>
+				<!-- deploy dependency jar to "shared", "common" or "server": use <war.bundle>true</war.bundle> to include in war -->
+				<deploy.target>shared</deploy.target>
+			</properties>
+		</dependency>
+
 		<dependency>
 			<groupId>servletapi</groupId>
 			<artifactId>servletapi</artifactId>
Index: master/jsf-sun-tool.xml
===================================================================
--- master/jsf-sun-tool.xml	(revision 19792)
+++ master/jsf-sun-tool.xml	(revision 19793)
@@ -53,7 +53,7 @@
 		<dependency>
 			<groupId>commons-fileupload</groupId>
 			<artifactId>commons-fileupload</artifactId>
-			<version>1.0</version>
+			<version>1.1.1</version>
 		</dependency>
 		
 		<dependency>
Index: osp/jsf/widgets/project.xml
===================================================================
--- osp/jsf/widgets/project.xml	(revision 19792)
+++ osp/jsf/widgets/project.xml	(revision 19793)
@@ -84,7 +84,7 @@
      <dependency>
       <groupId>commons-fileupload</groupId>
       <artifactId>commons-fileupload</artifactId>
-      <version>1.0</version>
+      <version>1.1.1</version>
       <url>http://jakarta.apache.org/commons/fileupload/</url>
     </dependency>
 
Index: osp/wizard/tool/project.xml
===================================================================
--- osp/wizard/tool/project.xml	(revision 19792)
+++ osp/wizard/tool/project.xml	(revision 19793)
@@ -175,7 +175,7 @@
         <dependency>
             <groupId>commons-fileupload</groupId>
             <artifactId>commons-fileupload</artifactId>
-            <version>1.0</version>
+            <version>1.1.1</version>
         </dependency>
         
         <dependency>
