svn diff -r31353:31354 https://source.sakaiproject.org/svn
Index: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java
===================================================================
--- site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 31353)
+++ site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 31354)
@@ -4586,7 +4586,8 @@
 					.getAttribute(STATE_TERM_SELECTED);
 			String sectionEid = sectionFieldManager.getSectionEid(t.getEid(),
 					(List) multiCourseInputs.get(0));
-			String title = "";
+			// default title
+			String title = sectionEid;
 			try {
 				title = cms.getSection(sectionEid).getTitle();
 			} catch (Exception e) {
svn diff -r31363:31364 https://source.sakaiproject.org/svn
Index: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java
===================================================================
--- site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 31363)
+++ site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 31364)
@@ -7881,7 +7881,7 @@
 							participant.removeable = false;
 							participant.role = member.getRole()!=null?member.getRole().getId():"";
 							participant.section = providerCourseEid;
-							participant.uniqname = user.getEid();
+							participant.uniqname = user.getId();
 							participants.add(participant);
 						}
 					} catch (UserNotDefinedException exception) {
svn diff -r31365:31366 https://source.sakaiproject.org/svn
Index: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java
===================================================================
--- site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 31365)
+++ site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 31366)
@@ -7873,14 +7873,13 @@
 						{
 							// add provided participant
 							Participant participant = new Participant();
-							participant.course = cms.getSection(providerCourseEid).getTitle();
 							participant.credits = e.getCredits();
 							participant.name = user.getSortName();
 							participant.providerRole = member.getRole()!=null?member.getRole().getId():"";
 							participant.regId = "";
 							participant.removeable = false;
 							participant.role = member.getRole()!=null?member.getRole().getId():"";
-							participant.section = providerCourseEid;
+							participant.section = cms.getSection(providerCourseEid).getTitle();
 							participant.uniqname = user.getId();
 							participants.add(participant);
 						}
@@ -10503,12 +10502,12 @@
 				// sort by whether the site is joinable or not
 				String s1 = null;
 				if (o1.getClass().equals(Participant.class)) {
-					s1 = ((Participant) o1).getCourse();
+					s1 = ((Participant) o1).getSection();
 				}
 
 				String s2 = null;
 				if (o2.getClass().equals(Participant.class)) {
-					s2 = ((Participant) o2).getCourse();
+					s2 = ((Participant) o2).getSection();
 				}
 
 				result = compareString(s1, s2);
@@ -10761,9 +10760,6 @@
 		/** The member credits */
 		protected String credits = NULL_STRING;
 
-		/** The course */
-		public String course = NULL_STRING;
-
 		/** The section */
 		public String section = NULL_STRING;
 
@@ -10798,10 +10794,6 @@
 			return credits;
 		} // getCredits
 
-		public String getCourse() {
-			return course;
-		} // getCourse
-
 		public String getSection() {
 			return section;
 		} // getSection
Index: site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-siteInfo-list.vm
===================================================================
--- site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-siteInfo-list.vm	(revision 31365)
+++ site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-siteInfo-list.vm	(revision 31366)
@@ -336,13 +336,9 @@
 						</td>
 						#if ($!isCourseSite)
 							<td headers="coursename">
-								#set($partCourse = "")
-								#if ($participant.getCourse())
-									#set($partCourse = $participant.getCourse())
-								#end
 								#set($partSection = "")
 								#set($partSection = $participant.getSection())
-								$!partCourse $!partSection
+								$!partSection
 							</td>
 							<td headers="id">
 								#set($partId = "")
