--- site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	2007-06-07 15:06:59.000000000 -0400
+++ site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	2007-06-07 15:07:16.000000000 -0400
@@ -4584,13 +4584,11 @@
 				&& multiCourseInputs.size() > 0) {
 			AcademicSession t = (AcademicSession) state
 					.getAttribute(STATE_TERM_SELECTED);
-			String courseId = sectionFieldManager.getSectionEid(t.getEid(),
+			String sectionEid = sectionFieldManager.getSectionEid(t.getEid(),
 					(List) multiCourseInputs.get(0));
 			String title = "";
 			try {
-				// instead of making up a name, Josh & I decided to use the
-				// section.eid instead
-				title = courseId;
+				title = cms.getSection(sectionEid).getTitle();
 			} catch (Exception e) {
 				// ignore
 			}
