Index: portal/portal-util/util/src/java/org/sakaiproject/portal/util/ErrorReporter.java
===================================================================
--- portal/portal-util/util/src/java/org/sakaiproject/portal/util/ErrorReporter.java	(revision 35703)
+++ portal/portal-util/util/src/java/org/sakaiproject/portal/util/ErrorReporter.java	(working copy)
@@ -387,6 +387,7 @@
 		try
 		{
 			// headers
+		        res.setStatus(javax.servlet.http.HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
 			res.setContentType("text/html; charset=UTF-8");
 			res.addDateHeader("Expires", System.currentTimeMillis()
 					- (1000L * 60L * 60L * 24L * 365L));
