Index: osp/matrix/tool/src/webapp/WEB-INF/jsp/matrix/viewCell.jsp
===================================================================
--- osp/matrix/tool/src/webapp/WEB-INF/jsp/matrix/viewCell.jsp	(revision 48597)
+++ osp/matrix/tool/src/webapp/WEB-INF/jsp/matrix/viewCell.jsp	(working copy)
@@ -312,9 +312,8 @@
                             <osp:param name="current_form_id" value="${node.resource.id}" />
                             <osp:param name="sakai_helperSessionId" value="${cell.uniqueId}" />
                             </osp:url>"
-							title="<fmt:message key="edit"/>"> <img
-							src="/library/image/silk/application_form_edit.png"
-							alt="<fmt:message key="edit"/>" /> </a>
+							title="<fmt:message key="edit"/>">
+							<fmt:message key="edit"/></a>
                         |
                         <a
 							href="<osp:url value="osp.wizard.page.contents.helper/formDelete.osp">
@@ -323,9 +322,9 @@
                             <osp:param name="current_form_id" value="${node.id}" />
                             <osp:param name="submit" value="delete" />
                             </osp:url>"
-							title="<fmt:message key="delete"/>"><img
-							src="/library/image/silk/application_form_delete.png"
-							alt="<fmt:message key="delete"/>" /></a>
+							title="<fmt:message key="remove"/>">
+                     <fmt:message key="remove"/>
+                     </a>
 						<c:if
 							test="${((isWizard != 'true' && matrixCan.review) || (isWizard == 'true' && wizardCan.review)) && cell.scaffoldingCell.reviewDevice != null}">
                         |
@@ -342,8 +341,8 @@
                           <osp:param name="objectTitle" value="${objectTitle}" />
                           <osp:param name="itemId" value="${node.id}" />
                           <osp:param name="sakai_helperSessionId" value="${cell.uniqueId}" />
-                          </osp:url>"><osp:message
-							key="review" /></a>
+                           </osp:url>"
+                        title="<fmt:message key="review"/>"> <fmt:message key="review"/></a>
 					</c:if></div>
 					</td>
 					<td><c:out value="${node.technicalMetadata.owner.displayName}" />
@@ -471,9 +470,8 @@
 						   <osp:param name="submit" value="delete"/>
 						   </osp:url>"
 					onclick="javascript:stopEvents(event)"
-					title="<fmt:message key="delete"/>"> <img
-					src="/library/image/silk/page_white_delete.png"
-					alt="<fmt:message key="delete"/>" /> </a>
+					title="<fmt:message key="remove"/>"> 
+					<fmt:message key="remove"/></a>
 			</c:if> <c:if
 				test="${((isWizard != 'true' && matrixCan.review) || (isWizard == 'true' && wizardCan.review)) && cell.scaffoldingCell.reviewDevice != null}">
 				<a
@@ -607,9 +605,8 @@
 			<img src='/library/image/silk/application_form.gif' border='0'
 				hspace='0' alt="" />
 			<c:out value="${reflections[0].reviewContentNode.displayName}" />
-			<span class="itemAction"> <img
-				src="/library/image/silk/application_form_edit.png"
-				alt="<fmt:message key="edit"/>" /> <a
+			<span class="itemAction"> 
+         <a
 				href="<osp:url value="osp.review.processor.helper/reviewHelper.osp">
 						   <osp:param name="page_id" value="${cell.wizardPage.id}" />
 						   <osp:param name="org_theospi_portfolio_review_type" value="0" />
