svn diff -r33334:33335 https://source.sakaiproject.org/svn
Index: metaobj/metaobj-impl/api-impl/src/resources/org/sakaiproject/metaobj/shared/control/formView.xslt
===================================================================
--- metaobj/metaobj-impl/api-impl/src/resources/org/sakaiproject/metaobj/shared/control/formView.xslt	(revision 33334)
+++ metaobj/metaobj-impl/api-impl/src/resources/org/sakaiproject/metaobj/shared/control/formView.xslt	(revision 33335)
@@ -12,10 +12,17 @@
 				</title>
 				<script type="text/javascript" language="JavaScript" src="/library/js/headscripts.js">  // empty
 					block </script>
+				<script language="JavaScript">
+					<![CDATA[
+						// in case the parent iframe id has whitespace, trim so that IE can use it.
+						function trim(s){
+						if((s==null)||(typeof(s)!='string')||!s.length)return'';return s.replace(/^\s+/,'').replace(/\s+$/,'')}
+					]]>
+				</script>
 				<link type="text/css" rel="stylesheet" media="all" href="/sakai-metaobj-tool/css/metaobj.css" />
 				<xsl:apply-templates select="css" />
 			</head>
-			<body onload="(window.frameElement) ? setMainFrameHeight(window.frameElement.id):''" class="formDisplay">
+			<body onload="(window.frameElement) ? setMainFrameHeight(trim(window.frameElement.id)):''" class="formDisplay">
 				<div class="portletBodyForm" style="padding:1em 0">
 					<xsl:apply-templates select="formData/artifact/schema/element">
 						<xsl:with-param name="currentParent" select="formData/artifact/structuredData" />
