Index: syllabus/syllabus-app/src/java/org/sakaiproject/tool/syllabus/SyllabusTool.java
===================================================================
--- syllabus/syllabus-app/src/java/org/sakaiproject/tool/syllabus/SyllabusTool.java	(revision 13976)
+++ syllabus/syllabus-app/src/java/org/sakaiproject/tool/syllabus/SyllabusTool.java	(revision 13977)
@@ -1489,7 +1489,7 @@
         session.getAttribute(FilePickerHelper.FILE_PICKER_ATTACHMENTS) != null) 
     {
       List refs = (List)session.getAttribute(FilePickerHelper.FILE_PICKER_ATTACHMENTS);
-      Reference ref = (Reference)refs.get(0);
+      Reference ref = null;
       
       for(int i=0; i<refs.size(); i++)
       {
@@ -1713,7 +1713,7 @@
         session.getAttribute(FilePickerHelper.FILE_PICKER_ATTACHMENTS) != null) 
     {
       List refs = (List)session.getAttribute(FilePickerHelper.FILE_PICKER_ATTACHMENTS);
-      Reference ref = (Reference)refs.get(0);
+      Reference ref = null;
       
       for(int i=0; i<refs.size(); i++)
       {
