--- db/db-util/storage/src/java/org/sakaiproject/util/BaseDbFlatStorage.original.java	2008-07-02 15:35:47.000000000 -0400
+++ db/db-util/storage/src/java/org/sakaiproject/util/BaseDbFlatStorage.java	2008-07-02 15:40:19.000000000 -0400
@@ -32,6 +32,9 @@
 import java.util.Properties;
 import java.util.Vector;
 
+import java.io.StringWriter;
+import java.io.PrintWriter;
+
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.sakaiproject.db.api.SqlReader;
@@ -1080,10 +1083,14 @@
 						}
 					}
 */
+					if (M_log.isDebugEnabled()) {
+					    M_log.debug("CACHE HIT cacheKey="+cacheKey+" props="+props);
+					    Throwable t = new Throwable();
+					    StringWriter stsw  = new StringWriter();
+					    t.printStackTrace(new PrintWriter(stsw));
+					    M_log.debug(stsw.toString());
+					}
 
-System.out.println("CACHE HIT cacheKey="+cacheKey+" props="+props);
-Throwable t = new Throwable();
-t.printStackTrace();
 					return;
 				}
 			}
@@ -1123,7 +1130,8 @@
 
 		if ( myCache != null )
 		{
-System.out.println("CACHE PUT cacheKey="+cacheKey+" props="+props);
+		    if (M_log.isDebugEnabled()) 
+			M_log.debug("CACHE PUT cacheKey="+cacheKey+" props="+props);
 			myCache.put(new Element(cacheKey,props));
 		}
 	}
@@ -1235,7 +1243,8 @@
 
 		Cache myCache = getCache(table);
 		String cacheKey = table + ":" + idField + ":" + id;
-System.out.println("CACHE REMOVE cacheKey="+cacheKey+" cache="+myCache);
+		if (M_log.isDebugEnabled()) 
+		    M_log.debug("CACHE REMOVE cacheKey="+cacheKey+" cache="+myCache);
 		if ( myCache != null )
 		{
 			myCache.remove(cacheKey);
