svn diff -r32930:32931 https://source.sakaiproject.org/svn
Index: site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourse.vm
===================================================================
--- site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourse.vm	(revision 32930)
+++ site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourse.vm	(revision 32931)
@@ -168,7 +168,7 @@
 			<label for="selectTerm">
 				$tlang.getString("nscourse.acad2")
 			</label>	
-			<select name="selectTerm" id="selectTerm" onchange="javascript:document.addCourseForm.option.value='change';document.addCourseForm.submit();">
+			<select name="selectTerm" id="selectTerm" onchange="javascript:document.getElementById('option').value='change';document.addCourseForm.submit();">
 				#foreach($t in $termList)
 					<option value ="$t.eid"
 						#if($t.eid == $term.eid)
