Index: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java
===================================================================
*** site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 33000)
--- site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 33001)
***************
*** 218,228 ****
  			"-siteInfo-importMtrlCopyConfirm",
  			"-siteInfo-importMtrlCopyConfirmMsg", // 48
  			"-siteInfo-group", // 49
  			"-siteInfo-groupedit", // 50
  			"-siteInfo-groupDeleteConfirm", // 51,
! 			"", // 52 - no template; used by daisyf. for remove course section
  			"-findCourse" // 53
  	};
  
  	/** Name of state attribute for Site instance id */
  	private static final String STATE_SITE_INSTANCE_ID = "site.instance.id";
--- 218,228 ----
  			"-siteInfo-importMtrlCopyConfirm",
  			"-siteInfo-importMtrlCopyConfirmMsg", // 48
  			"-siteInfo-group", // 49
  			"-siteInfo-groupedit", // 50
  			"-siteInfo-groupDeleteConfirm", // 51,
! 			"",
  			"-findCourse" // 53
  	};
  
  	/** Name of state attribute for Site instance id */
  	private static final String STATE_SITE_INSTANCE_ID = "site.instance.id";
***************
*** 2462,2474 ****
  				context.put("siteTitle", site.getTitle());
  				setTermListForContext(context, state, true); // true -> upcoming only
  
  				List providerCourseList = (List) state
  						.getAttribute(SITE_PROVIDER_COURSE_LIST);
! 				context.put("providerCourseList", providerCourseList);
! 				context.put("manualCourseList", state
! 						.getAttribute(SITE_MANUAL_COURSE_LIST));
  
  				AcademicSession t = (AcademicSession) state
  						.getAttribute(STATE_TERM_SELECTED);
  				context.put("term", t);
  				if (t != null) {
--- 2462,2472 ----
  				context.put("siteTitle", site.getTitle());
  				setTermListForContext(context, state, true); // true -> upcoming only
  
  				List providerCourseList = (List) state
  						.getAttribute(SITE_PROVIDER_COURSE_LIST);
! 				coursesIntoContext(state, context, site);
  
  				AcademicSession t = (AcademicSession) state
  						.getAttribute(STATE_TERM_SELECTED);
  				context.put("term", t);
  				if (t != null) {
***************
*** 2609,2636 ****
  			if (state.getAttribute(STATE_CM_REQUESTED_SECTIONS) != null) {
  				List l = (List) state
  						.getAttribute(STATE_CM_REQUESTED_SECTIONS);
  				context.put("requestedSections", l);
  			}
- 			
- 			// v2.4 - added & modified by daisyf
- 			if (courseManagementIsImplemented() && state.getAttribute(STATE_TERM_COURSE_LIST) != null)
- 			{
- 				// back to the list view of sections
- 				context.put("back", "36");
- 			} else {
- 				context.put("back", "1");
- 			}
  
  			if (site == null) {
  				if (state.getAttribute(STATE_AUTO_ADD) != null) {
  					context.put("autoAdd", Boolean.TRUE);
  					// context.put("back", "36");
- 				} else {
- 					context.put("back", "1");
  				}
  			}
  			context.put("isFutureTerm", state
  					.getAttribute(STATE_FUTURE_TERM_SELECTED));
  			context.put("weeksAhead", ServerConfigurationService.getString(
  					"roster.available.weeks.before.term.start", "0"));
  			return (String) getContext(data).get("template") + TEMPLATE[37];
--- 2607,2638 ----
  			if (state.getAttribute(STATE_CM_REQUESTED_SECTIONS) != null) {
  				List l = (List) state
  						.getAttribute(STATE_CM_REQUESTED_SECTIONS);
  				context.put("requestedSections", l);
  			}
  
  			if (site == null) {
+ 				// v2.4 - added & modified by daisyf
+ 				if (courseManagementIsImplemented() && state.getAttribute(STATE_TERM_COURSE_LIST) != null)
+ 				{
+ 					// back to the list view of sections
+ 					context.put("back", "36");
+ 				} else {
+ 					context.put("back", "1");
+ 				}
  				if (state.getAttribute(STATE_AUTO_ADD) != null) {
  					context.put("autoAdd", Boolean.TRUE);
  					// context.put("back", "36");
  				}
  			}
+ 			else
+ 			{
+ 				// editing site
+ 				context.put("back", "36");
+ 			}
+ 			
+ 			isFutureTermSelected(state);
  			context.put("isFutureTerm", state
  					.getAttribute(STATE_FUTURE_TERM_SELECTED));
  			context.put("weeksAhead", ServerConfigurationService.getString(
  					"roster.available.weeks.before.term.start", "0"));
  			return (String) getContext(data).get("template") + TEMPLATE[37];
***************
*** 3960,3989 ****
  				state.setAttribute(STATE_TERM_SELECTED, t);
  				if (t != null) {
  					List sections = prepareCourseAndSectionListing(userId, t
  							.getEid(), state);
  
! 					int weeks = 0;
! 					Calendar c = (Calendar) Calendar.getInstance().clone();
! 					try {
! 						weeks = Integer
! 								.parseInt(ServerConfigurationService
! 										.getString(
! 												"roster.available.weeks.before.term.start",
! 												"0"));
! 						c.add(Calendar.DATE, weeks * 7);
! 					} catch (Exception ignore) {
! 					}
! 
! 					if (c.getTimeInMillis() < t.getStartDate().getTime()) {
! 						// if a future term is selected
! 						state.setAttribute(STATE_FUTURE_TERM_SELECTED,
! 								Boolean.TRUE);
! 					} else {
! 						state.setAttribute(STATE_FUTURE_TERM_SELECTED,
! 								Boolean.FALSE);
! 					}
  
  					if (sections != null && sections.size() > 0) {
  						state.setAttribute(STATE_TERM_COURSE_LIST, sections);
  						state.setAttribute(STATE_TEMPLATE_INDEX, "36");
  						state.setAttribute(STATE_AUTO_ADD, Boolean.TRUE);
--- 3962,3972 ----
  				state.setAttribute(STATE_TERM_SELECTED, t);
  				if (t != null) {
  					List sections = prepareCourseAndSectionListing(userId, t
  							.getEid(), state);
  
! 					isFutureTermSelected(state);
  
  					if (sections != null && sections.size() > 0) {
  						state.setAttribute(STATE_TERM_COURSE_LIST, sections);
  						state.setAttribute(STATE_TEMPLATE_INDEX, "36");
  						state.setAttribute(STATE_AUTO_ADD, Boolean.TRUE);
***************
*** 4042,4051 ****
--- 4025,4063 ----
  			state.setAttribute(SITE_CREATE_CURRENT_STEP, new Integer(1));
  		}
  
  	} // doSite_type
  
+ 	/**
+ 	 * Determine whether the selected term is considered of "future term"
+ 	 * @param state
+ 	 * @param t
+ 	 */
+ 	private void isFutureTermSelected(SessionState state) {
+ 		AcademicSession t = (AcademicSession) state.getAttribute(STATE_TERM_SELECTED);
+ 		int weeks = 0;
+ 		Calendar c = (Calendar) Calendar.getInstance().clone();
+ 		try {
+ 			weeks = Integer
+ 					.parseInt(ServerConfigurationService
+ 							.getString(
+ 									"roster.available.weeks.before.term.start",
+ 									"0"));
+ 			c.add(Calendar.DATE, weeks * 7);
+ 		} catch (Exception ignore) {
+ 		}
+ 
+ 		if (c.getTimeInMillis() < t.getStartDate().getTime()) {
+ 			// if a future term is selected
+ 			state.setAttribute(STATE_FUTURE_TERM_SELECTED,
+ 					Boolean.TRUE);
+ 		} else {
+ 			state.setAttribute(STATE_FUTURE_TERM_SELECTED,
+ 					Boolean.FALSE);
+ 		}
+ 	}
+ 
  	public void doChange_user(RunData data) {
  		doSite_type(data);
  	} // doChange_user
  
  	/**
***************
*** 4367,4376 ****
--- 4379,4411 ----
  		state.setAttribute(SORTED_ASC, Boolean.TRUE.toString());
  
  		state.setAttribute(STATE_TEMPLATE_INDEX, "49");
  
  	} // doMenu_group
+ 	
+ 	/**
+ 	 * 
+ 	 */
+ 	private void removeSection(SessionState state, ParameterParser params)
+ 	{
+ 		// v2.4 - added by daisyf
+ 		// RemoveSection - remove any selected course from a list of
+ 		// provider courses
+ 		// check if any section need to be removed
+ 		removeAnyFlagedSection(state, params);
+ 		
+ 		SiteInfo siteInfo = new SiteInfo();
+ 		if (state.getAttribute(STATE_SITE_INFO) != null) {
+ 			siteInfo = (SiteInfo) state.getAttribute(STATE_SITE_INFO);
+ 		}
+ 
+ 		List providerChosenList = (List) state
+ 				.getAttribute(STATE_ADD_CLASS_PROVIDER_CHOSEN);
+ 		collectNewSiteInfo(siteInfo, state, params, providerChosenList);
+ 		// next step
+ 		//state.setAttribute(SITE_CREATE_CURRENT_STEP, new Integer(2));
+ 	}
  
  	/**
  	 * dispatch to different functions based on the option value in the
  	 * parameter
  	 */
***************
*** 4385,4398 ****
  
  			String uniqname = StringUtil.trimToNull(params
  					.getString("uniqname"));
  			state.setAttribute(STATE_SITE_QUEST_UNIQNAME, uniqname);
  
! 			if (getStateSite(state) == null) {
! 				// creating new site
! 				updateSiteInfo(params, state);
! 			}
  
  			if (option.equalsIgnoreCase("add")) {
  
  				if (state.getAttribute(STATE_FUTURE_TERM_SELECTED) != null
  						&& !((Boolean) state
--- 4420,4430 ----
  
  			String uniqname = StringUtil.trimToNull(params
  					.getString("uniqname"));
  			state.setAttribute(STATE_SITE_QUEST_UNIQNAME, uniqname);
  
! 			updateSiteInfo(params, state);
  
  			if (option.equalsIgnoreCase("add")) {
  
  				if (state.getAttribute(STATE_FUTURE_TERM_SELECTED) != null
  						&& !((Boolean) state
***************
*** 4439,4451 ****
--- 4471,4517 ----
  			if (getStateSite(state) == null) {
  				doCancel_create(data);
  			} else {
  				doCancel(data);
  			}
+ 		} else if (option.equalsIgnoreCase("removeSection"))
+ 		{
+ 			// remove selected section
+ 			removeSection(state, params);
  		}
  
  	} // doManual_add_course
+ 	
+ 
+ 	/**
+ 	 * dispatch to different functions based on the option value in the
+ 	 * parameter
+ 	 */
+ 	public void doSite_information(RunData data) {
+ 		SessionState state = ((JetspeedRunData) data)
+ 				.getPortletSessionState(((JetspeedRunData) data).getJs_peid());
+ 		ParameterParser params = data.getParameters();
+ 
+ 		String option = params.getString("option");
+ 		if (option.equalsIgnoreCase("continue"))
+ 		{
+ 			doContinue(data);
+ 		} else if (option.equalsIgnoreCase("back")) {
+ 			doBack(data);
+ 		} else if (option.equalsIgnoreCase("cancel")) {
+ 			if (getStateSite(state) == null) {
+ 				doCancel_create(data);
+ 			} else {
+ 				doCancel(data);
+ 			}
+ 		} else if (option.equalsIgnoreCase("removeSection"))
+ 		{
+ 			// remove selected section
+ 			removeSection(state, params);
+ 		}
+ 
+ 	} // doSite_information
  
  	/**
  	 * read the input information of subject, course and section in the manual
  	 * site creation page
  	 */
***************
*** 4720,4739 ****
--- 4786,4825 ----
  	/**
  	 * handle with continue add new course site options
  	 * 
  	 */
  	public void doContinue_new_course(RunData data) {
+ 		SessionState state = ((JetspeedRunData) data)
+ 				.getPortletSessionState(((JetspeedRunData) data).getJs_peid());
+ 		ParameterParser params = data.getParameters();
+ 		
  		String option = data.getParameters().getString("option");
  		if (option.equals("continue")) {
  			doContinue(data);
  		} else if (option.equals("cancel")) {
  			doCancel_create(data);
  		} else if (option.equals("back")) {
  			doBack(data);
  		} else if (option.equals("cancel")) {
  			doCancel_create(data);
  		}
+ 		else if (option.equalsIgnoreCase("change")) {
+ 			// change term
+ 			String termId = params.getString("selectTerm");
+ 			AcademicSession t = cms.getAcademicSession(termId);
+ 			state.setAttribute(STATE_TERM_SELECTED, t);
+ 			isFutureTermSelected(state);
+ 		} else if (option.equalsIgnoreCase("cancel_edit")) {
+ 			// cancel
+ 			state.removeAttribute(STATE_TERM_SELECTED);
+ 			removeAddClassContext(state);
+ 			state.setAttribute(STATE_TEMPLATE_INDEX, "43");
+ 		} else if (option.equalsIgnoreCase("add")) {
+ 			isFutureTermSelected(state);
+ 			// continue
+ 			doContinue(data);
+ 		}
  	} // doContinue_new_course
  
  	/**
  	 * doBack is called when "eventSubmit_doBack" is in the request parameters
  	 * Pass parameter to actionForTemplate to request action for backward
***************
*** 5975,6039 ****
  		state.setAttribute(STATE_TEMPLATE_INDEX, "36");
  
  	} // doMenu_siteInfo_addClass
  
  	/**
- 	 * first step of adding class
- 	 */
- 	public void doAdd_class_select(RunData data) {
- 		SessionState state = ((JetspeedRunData) data)
- 				.getPortletSessionState(((JetspeedRunData) data).getJs_peid());
- 		ParameterParser params = data.getParameters();
- 		String option = params.getString("option");
- 		if (option.equalsIgnoreCase("change")) {
- 			// change term
- 			String termId = params.getString("selectTerm");
- 			AcademicSession t = cms.getAcademicSession(termId);
- 			state.setAttribute(STATE_TERM_SELECTED, t);
- 		} else if (option.equalsIgnoreCase("cancel")) {
- 			// cancel
- 			state.removeAttribute(STATE_TERM_SELECTED);
- 			removeAddClassContext(state);
- 			state.setAttribute(STATE_TEMPLATE_INDEX, "43");
- 		} else if (option.equalsIgnoreCase("add")) {
- 			String userId = UserDirectoryService.getCurrentUser().getEid();
- 			AcademicSession t = (AcademicSession) state
- 					.getAttribute(STATE_TERM_SELECTED);
- 			if (t != null) {
- 				List courses = prepareCourseAndSectionListing(userId, t
- 						.getEid(), state);
- 
- 				// future term? roster information is not available yet?
- 				int weeks = 0;
- 				try {
- 					weeks = Integer.parseInt(ServerConfigurationService
- 							.getString(
- 									"roster.available.weeks.before.term.start",
- 									"0"));
- 				} catch (Exception ignore) {
- 					M_log.warn(ignore.getMessage());
- 				}
- 				if ((courses == null || courses != null && courses.size() == 0)
- 						&& System.currentTimeMillis() + weeks * 7 * 24 * 60
- 								* 60 * 1000 < t.getStartDate().getTime()) {
- 					// if a future term is selected
- 					state
- 							.setAttribute(STATE_FUTURE_TERM_SELECTED,
- 									Boolean.TRUE);
- 				} else {
- 					state.setAttribute(STATE_FUTURE_TERM_SELECTED,
- 							Boolean.FALSE);
- 				}
- 			}
- 
- 			// continue
- 			doContinue(data);
- 		}
- 
- 	} // doAdd_class_select
- 
- 	/**
  	 * doMenu_siteInfo_duplicate
  	 */
  	public void doMenu_siteInfo_duplicate(RunData data) {
  		SessionState state = ((JetspeedRunData) data)
  				.getPortletSessionState(((JetspeedRunData) data).getJs_peid());
--- 6061,6070 ----
***************
*** 7253,7275 ****
  				}
  				// next step
  				state.setAttribute(SITE_CREATE_CURRENT_STEP, new Integer(2));
  			}
  			break;
- 		case 52:
- 			// v2.4 - added by daisyf
- 			// RemoveSection - remove any selected course from a list of
- 			// provider courses
- 			// check if any section need to be removed
- 			removeAnyFlagedSection(state, params);
- 
- 			List providerChosenList = (List) state
- 					.getAttribute(STATE_ADD_CLASS_PROVIDER_CHOSEN);
- 			collectNewSiteInfo(siteInfo, state, params, providerChosenList);
- 			// next step
- 			state.setAttribute(SITE_CREATE_CURRENT_STEP, new Integer(2));
- 			break;
  		case 38:
  			break;
  		case 39:
  			break;
  		case 42:
--- 7284,7293 ----
***************
*** 7412,7421 ****
--- 7430,7440 ----
  		state.removeAttribute(STATE_AUTO_ADD);
  		state.removeAttribute(SITE_CREATE_TOTAL_STEPS);
  		state.removeAttribute(SITE_CREATE_CURRENT_STEP);
  		state.removeAttribute(STATE_IMPORT_SITE_TOOL);
  		state.removeAttribute(STATE_IMPORT_SITES);
+ 		sitePropertiesIntoState(state);
  
  	} // removeAddClassContext
  
  	private void updateCourseClasses(SessionState state, List notifyClasses,
  			List requestClasses) {
***************
*** 9583,9592 ****
--- 9602,9612 ----
  			siteInfo.infoUrl = site.getInfoUrl();
  			siteInfo.joinable = site.isJoinable();
  			siteInfo.joinerRole = site.getJoinerRole();
  			siteInfo.published = site.isPublished();
  			siteInfo.include = site.isPubView();
+ 			siteInfo.additional = "";
  			siteInfo.short_description = site.getShortDescription();
  			state.setAttribute(STATE_SITE_TYPE, siteInfo.site_type);
  
  			state.setAttribute(STATE_SITE_INFO, siteInfo);
  		} catch (Exception e) {
***************
*** 11674,11689 ****
  	private void removeAnyFlagedSection(SessionState state,
  			ParameterParser params) {
  		List all = new ArrayList();
  		List providerCourseList = (List) state
  				.getAttribute(STATE_ADD_CLASS_PROVIDER_CHOSEN);
! 		if (providerCourseList != null) {
  			all.addAll(providerCourseList);
  		}
  		List manualCourseList = (List) state
  				.getAttribute(SITE_MANUAL_COURSE_LIST);
! 		if (manualCourseList != null) {
  			all.addAll(manualCourseList);
  		}
  		
  		for (int i = 0; i < all.size(); i++) {
  			String eid = (String) all.get(i);
--- 11694,11709 ----
  	private void removeAnyFlagedSection(SessionState state,
  			ParameterParser params) {
  		List all = new ArrayList();
  		List providerCourseList = (List) state
  				.getAttribute(STATE_ADD_CLASS_PROVIDER_CHOSEN);
! 		if (providerCourseList != null && providerCourseList.size() > 0) {
  			all.addAll(providerCourseList);
  		}
  		List manualCourseList = (List) state
  				.getAttribute(SITE_MANUAL_COURSE_LIST);
! 		if (manualCourseList != null && manualCourseList.size() > 0) {
  			all.addAll(manualCourseList);
  		}
  		
  		for (int i = 0; i < all.size(); i++) {
  			String eid = (String) all.get(i);
***************
*** 11785,11799 ****
  				state.removeAttribute(STATE_SITE_QUEST_UNIQNAME);
  
  				if (getStateSite(state) != null) {
  					// if revising a site, go to the confirmation
  					// page of adding classes
! 					state.setAttribute(STATE_TEMPLATE_INDEX, "44");
  				} else {
  					// if creating a site, go the the site
  					// information entry page
! 					state.setAttribute(STATE_TEMPLATE_INDEX, "2");
  				}
  			}
  		}
  	}
  
--- 11805,11819 ----
  				state.removeAttribute(STATE_SITE_QUEST_UNIQNAME);
  
  				if (getStateSite(state) != null) {
  					// if revising a site, go to the confirmation
  					// page of adding classes
! 					//state.setAttribute(STATE_TEMPLATE_INDEX, "37");
  				} else {
  					// if creating a site, go the the site
  					// information entry page
! 					//state.setAttribute(STATE_TEMPLATE_INDEX, "2");
  				}
  			}
  		}
  	}
  
Index: site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourseManual.vm
===================================================================
*** site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourseManual.vm	(revision 33000)
--- site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourseManual.vm	(revision 33001)
***************
*** 2,22 ****
  <script type="text/javascript">
  	function submitRemoveSection(index){
  		id = "removeSection"+index;
  		removeSection = document.getElementById(id);
  		removeSection.value="true";
! 
! 		document.removeSectionForm.submit();
  		return false;
  }
  </script>
  
  <div class="portletBody">
- 	<form name = "removeSectionForm" method="post" action="#toolForm($action)">
- 		<input type="hidden" name="eventSubmit_doContinue" value="$tlang.getString('gen.continue')" />
- 		<input type="hidden" name="back" value="1" id="back" />
- 		<input type="hidden" name="template-index" value="52"  />
  	
  	##if($menu)#toolbar($menu)#end<br/><br/>
  		#if (!$!site)
  			<h3>$tlang.getString("man.creat")</h3>
  			<p class="step">
--- 2,18 ----
  <script type="text/javascript">
  	function submitRemoveSection(index){
  		id = "removeSection"+index;
  		removeSection = document.getElementById(id);
  		removeSection.value="true";
! 		document.getElementById("option").value="removeSection";
! 		document.requestCourseForm.submit();
  		return false;
  }
  </script>
  
  <div class="portletBody">
  	
  	##if($menu)#toolbar($menu)#end<br/><br/>
  		#if (!$!site)
  			<h3>$tlang.getString("man.creat")</h3>
  			<p class="step">
***************
*** 27,57 ****
  		#end
  		#if ($alertMessage)
  			<div class="alertMessage">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div>
  			<div class="clear"></div>
  		#end
! 			#if ($!site)
  				<p class="instruction">
  				$tlang.getString("nscourse.already")
  				</p>	
  				#foreach($id in $!providerCourseList)
  					<p class="indnt2">
! 						$id 
! 						<!-- Remove Section -->
! 						(<a href="#" onclick="javascript:submitRemoveSection('$id');">Remove</a>)
! 						<input type="hidden" id="removeSection$item" name="removeSection$id" value="" />
  					</p>
  				#end
  				#foreach($id in $!manualCourseList)
  					<p class="indnt2">
  						$id <span class="highlight">$tlang.getString("man.requested")</span>
- 						<!-- Remove Section -->
- 						(<a href="#" onclick="javascript:submitRemoveSection('$id');">Remove</a>)
- 						<input type="hidden" id="removeSection$id" name="removeSection$item" value="" />
  					</p>
  				#end
- 				
  			#end
  				#if ($!selectedProviderCourse)
  				<p class="instruction">$tlang.getString("man.sofar") #if($!site) $tlang.getString("man.toaddto") #else $tlang.getString("java.for") #end $tlang.getString("man.this")</p> 
  				<ul>
  					#foreach ($item in $!selectedProviderCourse)
--- 23,50 ----
  		#end
  		#if ($alertMessage)
  			<div class="alertMessage">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div>
  			<div class="clear"></div>
  		#end
! 		
! 		<form name = "requestCourseForm" method = "post" action="#toolForm("$action")">
! 			<input type="hidden" name="option" id="option" value="add" />
! 			<input type="hidden" name="eventSubmit_doManual_add_course" value="x" />
! 			#if ($!site && ($!providerCourseList.size() > 0 || $!manualCourseList.size()>0 ))
  				<p class="instruction">
  				$tlang.getString("nscourse.already")
  				</p>	
  				#foreach($id in $!providerCourseList)
  					<p class="indnt2">
! 						$id
  					</p>
  				#end
  				#foreach($id in $!manualCourseList)
  					<p class="indnt2">
  						$id <span class="highlight">$tlang.getString("man.requested")</span>
  					</p>
  				#end
  			#end
  				#if ($!selectedProviderCourse)
  				<p class="instruction">$tlang.getString("man.sofar") #if($!site) $tlang.getString("man.toaddto") #else $tlang.getString("java.for") #end $tlang.getString("man.this")</p> 
  				<ul>
  					#foreach ($item in $!selectedProviderCourse)
***************
*** 71,85 ****
  							<input type="hidden" id="removeSection$section.eid" name="removeSection$section.eid" value="" />
  					</li>
  					#end
  				</ul>
  				#end
- 	</form>
  				
- 	<form name = "requestCourseForm" method = "post" action="#toolForm("$action")">
- 				<input type="hidden" name="option" value="add" />
- 				<input type="hidden" name="eventSubmit_doManual_add_course" value="x" />
  				<p class="instruction">
  					#if (!$!isFutureTerm)
  						$tlang.getString("man.please_enter")
  					#else
  						$tlang.getString("man.enter2") <span class="reqStarInline">*</span> $tlang.getString("man.means")
--- 64,74 ----
Index: site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteInformation.vm
===================================================================
*** site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteInformation.vm	(revision 33000)
--- site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteInformation.vm	(revision 33001)
***************
*** 18,30 ****
  	  	return result;
  	}
  	
  	function submitRemoveSection(index){
  		id = "removeSection"+index;
!     	removeSection = document.getElementById(id);
  		removeSection.value="true";
! 		document.removeSectionForm.submit();
  		return false;
  }
  	
  </script>
  <div class="portletBody">
--- 18,31 ----
  	  	return result;
  	}
  	
  	function submitRemoveSection(index){
  		id = "removeSection"+index;
!   removeSection = document.getElementById(id);
  		removeSection.value="true";
! 		document.getElementById("option").value="removeSection";
! 		document.siteInfoForm.submit();
  		return false;
  }
  	
  </script>
  <div class="portletBody">
***************
*** 53,67 ****
  	#if ($alertMessage)
  		<div class="alertMessage">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div>
  		<div class="clear"></div>
  	#end
  	
! 	<form name = "removeSectionForm" method="post" action="#toolForm($action)">
! 		<input type="hidden" name="eventSubmit_doContinue" value="$tlang.getString('gen.continue')" />
! 		<input type="hidden" name="back" value="1" />
! 		<input type="hidden" name="template-index" value="52" />
! 	
  	#if ($isCourseSite)
  		#if ($!selectedProviderCourse)
  		$tlang.getString("sinfo.youare")
  		<ul>
  			#foreach ($item in $!selectedProviderCourse)
--- 54,66 ----
  	#if ($alertMessage)
  		<div class="alertMessage">$tlang.getString("gen.alert") $validator.escapeHtml($alertMessage)</div>
  		<div class="clear"></div>
  	#end
  	
! 	<form name="siteInfoForm" action="#toolForm("$action")" method="post">
! 		<input type="hidden" name="option" id="option" value="continue" />
! 		<input type="hidden" name="eventSubmit_doSite_information" value="x" />	
  	#if ($isCourseSite)
  		#if ($!selectedProviderCourse)
  		$tlang.getString("sinfo.youare")
  		<ul>
  			#foreach ($item in $!selectedProviderCourse)
***************
*** 104,116 ****
  						</li>
  				#end
  			</ul>
  		#end
  	#end
- 	</form>
  	
- 	<form action="#toolForm("$action")" method="post">	
  	<p class="instruction">
  		$tlang.getString("sinfo.enter3")#if ($!isProjectSite) $tlang.getString("sinfo.simplea") <span class ="reqStarInline">*</span> $tlang.getString("sinfo.means") #end
  	</p>
  		<p class="shorttext required">
  			#if ($!titleEditableSiteType.contains($!type))
--- 103,113 ----
***************
*** 192,205 ****
  		</p>
  		<input type="hidden" name="back" value="$back" />
  		<input type="hidden" name="template-index" value="2" />
  		<input type="hidden" name="continue" value="3" />
  		<p class="act">
! 			<input id="continueButton" type="submit" accesskey="s" class="active" name="eventSubmit_doContinue" value="$tlang.getString('gen.continue')" />
! 			#if ($!siteTypes.size() > 1)
! 				<input type="submit" accesskey="b" name="eventSubmit_doBack" value="$tlang.getString('gen.back')" />
! 			#end
! 			<input type="submit" accesskey="x" name="eventSubmit_doCancel_create" value="$tlang.getString('gen.cancel')" />
  		</p>
  	</form>
  </div>
  
--- 189,213 ----
  		</p>
  		<input type="hidden" name="back" value="$back" />
  		<input type="hidden" name="template-index" value="2" />
  		<input type="hidden" name="continue" value="3" />
  		<p class="act">
! 				<input type="button" name="continueButton" id="continueButton"
! 						value="$tlang.getString('gen.continue')" 
! 						class="active" 
! 						accesskey="s" 
! 						onclick="document.siteInfoForm.option.value='continue'; document.siteInfoForm.submit(); return false;" />
! 				#if ($!siteTypes.size() > 1)
! 					<input type="button" name="Back" 
! 							value="$tlang.getString('gen.back')"
! 							accesskey="b"
! 							onclick="document.siteInfoForm.option.value='back'; document.siteInfoForm.submit(); return false;" />
! 				#end
! 				<input type="button" name="Cancel" 
! 						value="$tlang.getString('gen.cancel')"
! 						accesskey="x"
! 						onclick="document.siteInfoForm.option.value='cancel'; document.siteInfoForm.submit(); return false;" />
! 			</p>
  		</p>
  	</form>
  </div>
  
Index: site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourse.vm
===================================================================
*** site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourse.vm	(revision 33000)
--- site-manage/site-manage-tool/tool/src/webapp/vm/sitesetup/chef_site-newSiteCourse.vm	(revision 33001)
***************
*** 1,8 ****
  ##<!-- $Header: /cvs/sakai2/legacy/tools/src/webapp/vm/sitesetup/chef_site-newSiteCourse.vm,v 1.3 2005/05/25 17:42:01 gsilver.umich.edu Exp $ -->
  <script type="text/javascript">
! <!-- hide from non-JS browsers
  function toggle(toggleKey, begin, end) {
      var checkboxes = document.getElementsByName("providerCourseAdd");
      for (i=begin; i<end; i++){
          if (checkboxes[toggleKey].checked){
             checkboxes[i].checked = true;
--- 1,8 ----
  ##<!-- $Header: /cvs/sakai2/legacy/tools/src/webapp/vm/sitesetup/chef_site-newSiteCourse.vm,v 1.3 2005/05/25 17:42:01 gsilver.umich.edu Exp $ -->
  <script type="text/javascript">
! <!-- hide from non-JS browsersf
  function toggle(toggleKey, begin, end) {
      var checkboxes = document.getElementsByName("providerCourseAdd");
      for (i=begin; i<end; i++){
          if (checkboxes[toggleKey].checked){
             checkboxes[i].checked = true;
***************
*** 153,170 ****
  			#foreach($id in $!providerCourseList)
  				<li>
  						$id
  				</li>
  			#end
! 			#foreach($id in $!requestedCourseList)
  				<li>
  						$id
  				</li>
  			#end
  		</ul>
  
! 		 #if ($site)
  		<p class="shorttext  required">
  			<span class ="reqStar">*</span>
  			<label for="selectTerm">
  				$tlang.getString("nscourse.acad2")
  			</label>	
--- 153,170 ----
  			#foreach($id in $!providerCourseList)
  				<li>
  						$id
  				</li>
  			#end
! 			#foreach($id in $!manualCourseList)
  				<li>
  						$id
  				</li>
  			#end
  		</ul>
  
! 		#if ($site)
  		<p class="shorttext  required">
  			<span class ="reqStar">*</span>
  			<label for="selectTerm">
  				$tlang.getString("nscourse.acad2")
  			</label>	
***************
*** 178,189 ****
  					$t.title
  					</option>
  				#end
  			</select> 
  		</p>
! 		#else
! 		#if ($enableCourseCreationForUser )
  		<table>
  			<tr>
  			<td>
  				<b>$tlang.getString("nscourse.current_displaying_courses_sections_for_id")</b>
  			</td>
--- 178,189 ----
  					$t.title
  					</option>
  				#end
  			</select> 
  		</p>
! 		#end
! 	#if ($enableCourseCreationForUser )
  		<table>
  			<tr>
  			<td>
  				<b>$tlang.getString("nscourse.current_displaying_courses_sections_for_id")</b>
  			</td>
***************
*** 206,216 ****
  				</div>
  			</td>
  			</tr>
  		</table>
  		#end
- 		#end
  
  
  			#if ($!termCourseList)
  				#if ($!providerCourseList.size() != 0 || $!requestedCourseList.size() != 0)
  					<p class="instruction">$tlang.getString("nscourse.youmay")</p>
--- 206,215 ----
***************
*** 310,323 ****
  
  
  		<input id="index" type="hidden" name="template-index" value="36" />
  		<input type="hidden" name="back" value="1" />
  		<input type="hidden" name="option" id="option" value="x" />
! 
  			#if (!$site)
  				<!-- Add Not Listed Courses -->
- 				<input type="hidden" name="eventSubmit_doContinue_new_course" value="x" />
  				<a href="#" onclick="javascript:submitFindCourse();">$tlang.getString('nscourse.add_course_not_listed')</a>
  				<p class="act">
  				<input
  					class="active"
  					disabled="true"
--- 309,321 ----
  
  
  		<input id="index" type="hidden" name="template-index" value="36" />
  		<input type="hidden" name="back" value="1" />
  		<input type="hidden" name="option" id="option" value="x" />
! 		<input type="hidden" name="eventSubmit_doContinue_new_course" value="x" />
  			#if (!$site)
  				<!-- Add Not Listed Courses -->
  				<a href="#" onclick="javascript:submitFindCourse();">$tlang.getString('nscourse.add_course_not_listed')</a>
  				<p class="act">
  				<input
  					class="active"
  					disabled="true"
***************
*** 344,356 ****
  					value="$tlang.getString('gen.cancel')"
  					onclick="document.addCourseForm.option.value='cancel'; document.addCourseForm.submit(); return false;"
  					/>
  				</p>
  			#else
  				<p class="act">
  				<input type="hidden" name="option" id="option" value="x" />
- 				<input type="hidden" name="eventSubmit_doAdd_class_select" value="x" />
  				<input
  					disabled="true"
  					type="button" 
  					accesskey="s"
  					name="AddClass" 
--- 342,355 ----
  					value="$tlang.getString('gen.cancel')"
  					onclick="document.addCourseForm.option.value='cancel'; document.addCourseForm.submit(); return false;"
  					/>
  				</p>
  			#else
+ 				<!-- Add Not Listed Courses -->
+ 				<a href="#" onclick="javascript:submitFindCourse();">$tlang.getString('nscourse.add_course_not_listed')</a>
  				<p class="act">
  				<input type="hidden" name="option" id="option" value="x" />
  				<input
  					disabled="true"
  					type="button" 
  					accesskey="s"
  					name="AddClass" 
***************
*** 362,372 ****
  					type="button" 
  					accesskey="x"
  					name="Cancel" 
  					id="Cancel" 
  					value="$tlang.getString('gen.cancel')"
! 					onclick="document.getElementById('option').value='cancel'; document.addCourseForm.submit(); return false;"
  				/>
  				</p>
  			#end
  
  	</form>
--- 361,371 ----
  					type="button" 
  					accesskey="x"
  					name="Cancel" 
  					id="Cancel" 
  					value="$tlang.getString('gen.cancel')"
! 					onclick="document.getElementById('option').value='cancel_edit'; document.addCourseForm.submit(); return false;"
  				/>
  				</p>
  			#end
  
  	</form>
Index: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java
===================================================================
*** site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 33098)
--- site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 33099)
***************
*** 11809,11819 ****
  					// page of adding classes
  					//state.setAttribute(STATE_TEMPLATE_INDEX, "37");
  				} else {
  					// if creating a site, go the the site
  					// information entry page
! 					//state.setAttribute(STATE_TEMPLATE_INDEX, "2");
  				}
  			}
  		}
  	}
  
--- 11809,11819 ----
  					// page of adding classes
  					//state.setAttribute(STATE_TEMPLATE_INDEX, "37");
  				} else {
  					// if creating a site, go the the site
  					// information entry page
! 					state.setAttribute(STATE_TEMPLATE_INDEX, "2");
  				}
  			}
  		}
  	}
  
