Index: courier/courier-tool/tool/src/java/org/sakaiproject/courier/tool/CourierTool.java
===================================================================
--- courier/courier-tool/tool/src/java/org/sakaiproject/courier/tool/CourierTool.java	(revision 34435)
+++ courier/courier-tool/tool/src/java/org/sakaiproject/courier/tool/CourierTool.java	(working copy)
@@ -110,7 +110,7 @@
             
             // the 3rd part. if it has one dash then it's the add on from  \/ the head script:
             //    updateReq.open("GET", url + "/" + new Date().getTime() + "-" + Math.random() + "?auto=true", true);
-            if(parts.length >= 3 && parts[2].split("-").length != 1) {
+            if(parts.length >= 4 && parts[2].split("-").length != 1) {
                String subContextId = parts[2];
                // refresh our presence at the location (placement)
                if (M_log.isDebugEnabled()) M_log.debug("setting presence subcontext: " + subContextId);
