svn diff -r30766:30767 https://source.sakaiproject.org/svn
Index: site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java
===================================================================
--- site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 30766)
+++ site-manage/site-manage-tool/tool/src/java/org/sakaiproject/site/tool/SiteAction.java	(revision 30767)
@@ -3861,6 +3861,9 @@
 			state.setAttribute(SORTED_BY, SORTED_BY_PARTICIPANT_NAME);
 			state.setAttribute(SORTED_ASC, Boolean.TRUE.toString());
 		}
+		
+		// reset the paging info
+		resetPaging(state);
 
 		if (((String) state.getAttribute(STATE_SITE_MODE))
 				.equalsIgnoreCase(SITE_MODE_SITESETUP)) {
