svn diff -r18595:18596 https://source.sakaiproject.org/svn
Index: content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java
===================================================================
--- content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java	(revision 18595)
+++ content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java	(revision 18596)
@@ -4584,7 +4584,7 @@
 		
 		if(this.m_prioritySortEnabled)
 		{
-			((BasicGroupAwareEdit) edit).setPriority();
+			// ((BasicGroupAwareEdit) edit).setPriority();
 		}
 		
 		// update the properties for update
@@ -8471,7 +8471,7 @@
 					m_properties = new BaseResourcePropertiesEdit(element);
 					if(m_prioritySortEnabled)
 					{
-						setPriority();
+						// setPriority();
 					}
 				}
 				// look for groups 
svn diff -r19358:19359 https://source.sakaiproject.org/svn
Index: content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java
===================================================================
--- content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java	(revision 19358)
+++ content/content-impl/impl/src/java/org/sakaiproject/content/impl/BaseContentService.java	(revision 19359)
@@ -9243,7 +9243,7 @@
 					m_properties = new BaseResourcePropertiesEdit(element);
 					if(m_prioritySortEnabled)
 					{
-						setPriority();
+						// setPriority();
 					}
 				}
 				// look for groups
