svn diff -r20348:20349 https://source.sakaiproject.org/svn
Index: rwiki/rwiki-model/src/java/uk/ac/cam/caret/sakai/rwiki/model/RWikiObjectContentImpl.java
===================================================================
--- rwiki/rwiki-model/src/java/uk/ac/cam/caret/sakai/rwiki/model/RWikiObjectContentImpl.java	(revision 20348)
+++ rwiki/rwiki-model/src/java/uk/ac/cam/caret/sakai/rwiki/model/RWikiObjectContentImpl.java	(revision 20349)
@@ -60,9 +60,9 @@
 
 	public void setContent(String content)
 	{
-		this.content= content.replaceAll("[\\x00-\\x08\\x0b\\x0c\\x0e-\\x1f\\ud800-\\udfff\\uffff\\ufffe]", "");
 		// SAK-2470
 		if (this.content == null) this.content = "";
+		this.content= content.replaceAll("[\\x00-\\x08\\x0b\\x0c\\x0e-\\x1f\\ud800-\\udfff\\uffff\\ufffe]", "");
 	}
 
 }
